{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import json\n",
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "taxa_json = '../genome-properties/docs/release/hierarchy.json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open(taxa_json, 'r') as json_file:\n",
    "    hierarchy = json.load(json_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "properties = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "class GenomeProperty(object):\n",
    "    def __init__(self, gen_prop_id, name, parent=None, childern=[]):\n",
    "        self.gen_prop_id = gen_prop_id\n",
    "        self.name = name\n",
    "        self.parent = parent\n",
    "        self.childern = childern"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def parse_genome_property_json(gen_prop_dict, current_json_node, parent_property=None):\n",
    "    gen_prop_id = current_json_node['id']\n",
    "    name = current_json_node['name']\n",
    "    children_json_nodes = current_json_node['children']\n",
    "    \n",
    "    current_genome_property = GenomeProperty(gen_prop_id, name, parent=parent_property)\n",
    "    \n",
    "    child_genome_properties = []\n",
    "    for child_json in children_json_nodes:\n",
    "        child_genome_prop = parse_genome_property_json(gen_prop_dict, child_json, parent_property=current_genome_property)\n",
    "        child_genome_properties.append(child_genome_prop)\n",
    "            \n",
    "    current_genome_property.childern = child_genome_properties\n",
    "    gen_prop_dict[current_genome_property.gen_prop_id] = current_genome_property\n",
    "    \n",
    "    return current_genome_property"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "top_level_property = parse_genome_property_json(properties, hierarchy)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def build_parent_child_list(genome_property, parent_child_mapping_list):\n",
    "    parent_id = genome_property.gen_prop_id\n",
    "    for child in genome_property.childern:\n",
    "        parent_child_mapping_list.append((parent_id, child.gen_prop_id))\n",
    "        build_parent_child_list(child, parent_child_mapping_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "parent_child_table = []\n",
    "build_parent_child_list(top_level_property,parent_child_table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('GenProp0065', 'GenProp0017'),\n",
       " ('GenProp0017', 'GenProp0063'),\n",
       " ('GenProp0063', 'GenProp0077'),\n",
       " ('GenProp0077', 'GenProp0770'),\n",
       " ('GenProp0063', 'GenProp0126'),\n",
       " ('GenProp0126', 'GenProp0001'),\n",
       " ('GenProp0126', 'GenProp0037'),\n",
       " ('GenProp0126', 'GenProp0109'),\n",
       " ('GenProp0126', 'GenProp0111'),\n",
       " ('GenProp0126', 'GenProp0117'),\n",
       " ('GenProp0126', 'GenProp0118'),\n",
       " ('GenProp0126', 'GenProp0159'),\n",
       " ('GenProp0126', 'GenProp0160'),\n",
       " ('GenProp0126', 'GenProp0162'),\n",
       " ('GenProp0126', 'GenProp0163'),\n",
       " ('GenProp0126', 'GenProp0164'),\n",
       " ('GenProp0126', 'GenProp0178'),\n",
       " ('GenProp0126', 'GenProp0199'),\n",
       " ('GenProp0199', 'GenProp0125'),\n",
       " ('GenProp0125', 'GenProp0160'),\n",
       " ('GenProp0125', 'GenProp0787'),\n",
       " ('GenProp0125', 'GenProp0786'),\n",
       " ('GenProp0199', 'GenProp0193'),\n",
       " ('GenProp0199', 'GenProp0788'),\n",
       " ('GenProp0788', 'GenProp0160'),\n",
       " ('GenProp0063', 'GenProp0184'),\n",
       " ('GenProp0184', 'GenProp0001'),\n",
       " ('GenProp0184', 'GenProp0030'),\n",
       " ('GenProp0184', 'GenProp0036'),\n",
       " ('GenProp0184', 'GenProp0038'),\n",
       " ('GenProp0184', 'GenProp0057'),\n",
       " ('GenProp0184', 'GenProp0124'),\n",
       " ('GenProp0184', 'GenProp0136'),\n",
       " ('GenProp0184', 'GenProp0140'),\n",
       " ('GenProp0140', 'GenProp0137'),\n",
       " ('GenProp0140', 'GenProp0138'),\n",
       " ('GenProp0184', 'GenProp0170'),\n",
       " ('GenProp0184', 'GenProp0171'),\n",
       " ('GenProp0184', 'GenProp0250'),\n",
       " ('GenProp0184', 'GenProp0255'),\n",
       " ('GenProp0255', 'GenProp0217'),\n",
       " ('GenProp0255', 'GenProp0223'),\n",
       " ('GenProp0184', 'GenProp0261'),\n",
       " ('GenProp0261', 'GenProp0250'),\n",
       " ('GenProp0261', 'GenProp0253'),\n",
       " ('GenProp0261', 'GenProp0254'),\n",
       " ('GenProp0184', 'GenProp0300'),\n",
       " ('GenProp0300', 'GenProp0144'),\n",
       " ('GenProp0144', 'GenProp1004'),\n",
       " ('GenProp0300', 'GenProp0146'),\n",
       " ('GenProp0300', 'GenProp0150'),\n",
       " ('GenProp0300', 'GenProp1004'),\n",
       " ('GenProp0184', 'GenProp0836'),\n",
       " ('GenProp0836', 'GenProp0058'),\n",
       " ('GenProp0836', 'GenProp0829'),\n",
       " ('GenProp0063', 'GenProp0185'),\n",
       " ('GenProp0185', 'GenProp0110'),\n",
       " ('GenProp0185', 'GenProp0187'),\n",
       " ('GenProp0063', 'GenProp0186'),\n",
       " ('GenProp0186', 'GenProp0055'),\n",
       " ('GenProp0186', 'GenProp0156'),\n",
       " ('GenProp0186', 'GenProp0168'),\n",
       " ('GenProp0186', 'GenProp0183'),\n",
       " ('GenProp0183', 'GenProp1032'),\n",
       " ('GenProp1032', 'GenProp0950'),\n",
       " ('GenProp0186', 'GenProp0297'),\n",
       " ('GenProp0063', 'GenProp0197'),\n",
       " ('GenProp0197', 'GenProp0132'),\n",
       " ('GenProp0197', 'GenProp0188'),\n",
       " ('GenProp0197', 'GenProp0189'),\n",
       " ('GenProp0063', 'GenProp0267'),\n",
       " ('GenProp0267', 'GenProp0147'),\n",
       " ('GenProp0063', 'GenProp0795'),\n",
       " ('GenProp0795', 'GenProp0183'),\n",
       " ('GenProp0183', 'GenProp1032'),\n",
       " ('GenProp1032', 'GenProp0950'),\n",
       " ('GenProp0795', 'GenProp0297'),\n",
       " ('GenProp0063', 'GenProp0838'),\n",
       " ('GenProp0838', 'GenProp0046'),\n",
       " ('GenProp0046', 'GenProp0047'),\n",
       " ('GenProp0046', 'GenProp0048'),\n",
       " ('GenProp0838', 'GenProp0136'),\n",
       " ('GenProp0838', 'GenProp0297'),\n",
       " ('GenProp0838', 'GenProp0299'),\n",
       " ('GenProp0838', 'GenProp0836'),\n",
       " ('GenProp0836', 'GenProp0058'),\n",
       " ('GenProp0836', 'GenProp0829'),\n",
       " ('GenProp0017', 'GenProp0070'),\n",
       " ('GenProp0070', 'GenProp0023'),\n",
       " ('GenProp0070', 'GenProp0033'),\n",
       " ('GenProp0033', 'GenProp0630'),\n",
       " ('GenProp0070', 'GenProp0120'),\n",
       " ('GenProp0070', 'GenProp0131'),\n",
       " ('GenProp0131', 'GenProp0130'),\n",
       " ('GenProp0070', 'GenProp0243'),\n",
       " ('GenProp0243', 'GenProp0128'),\n",
       " ('GenProp0070', 'GenProp0298'),\n",
       " ('GenProp0017', 'GenProp0072'),\n",
       " ('GenProp0072', 'GenProp0118'),\n",
       " ('GenProp0072', 'GenProp0120'),\n",
       " ('GenProp0072', 'GenProp0143'),\n",
       " ('GenProp0072', 'GenProp0196'),\n",
       " ('GenProp0196', 'GenProp0030'),\n",
       " ('GenProp0196', 'GenProp0036'),\n",
       " ('GenProp0196', 'GenProp0140'),\n",
       " ('GenProp0140', 'GenProp0137'),\n",
       " ('GenProp0140', 'GenProp0138'),\n",
       " ('GenProp0196', 'GenProp0155'),\n",
       " ('GenProp0196', 'GenProp0182'),\n",
       " ('GenProp0196', 'GenProp0191'),\n",
       " ('GenProp0072', 'GenProp0202'),\n",
       " ('GenProp0202', 'GenProp0029'),\n",
       " ('GenProp0029', 'GenProp0633'),\n",
       " ('GenProp0029', 'GenProp0631'),\n",
       " ('GenProp0029', 'GenProp0632'),\n",
       " ('GenProp0072', 'GenProp0242'),\n",
       " ('GenProp0242', 'GenProp0190'),\n",
       " ('GenProp0242', 'GenProp0243'),\n",
       " ('GenProp0243', 'GenProp0128'),\n",
       " ('GenProp0072', 'GenProp0865'),\n",
       " ('GenProp0017', 'GenProp0133'),\n",
       " ('GenProp0133', 'GenProp0116'),\n",
       " ('GenProp0116', 'GenProp0115'),\n",
       " ('GenProp0115', 'GenProp0114'),\n",
       " ('GenProp0115', 'GenProp0198'),\n",
       " ('GenProp0115', 'GenProp0216'),\n",
       " ('GenProp0115', 'GenProp0227'),\n",
       " ('GenProp0115', 'GenProp0228'),\n",
       " ('GenProp0116', 'GenProp0198'),\n",
       " ('GenProp0133', 'GenProp0132'),\n",
       " ('GenProp0133', 'GenProp0185'),\n",
       " ('GenProp0185', 'GenProp0110'),\n",
       " ('GenProp0185', 'GenProp0187'),\n",
       " ('GenProp0133', 'GenProp0816'),\n",
       " ('GenProp0133', 'GenProp0817'),\n",
       " ('GenProp0133', 'GenProp0819'),\n",
       " ('GenProp0819', 'GenProp0262'),\n",
       " ('GenProp0017', 'GenProp0142'),\n",
       " ('GenProp0142', 'GenProp0055'),\n",
       " ('GenProp0142', 'GenProp0141'),\n",
       " ('GenProp0142', 'GenProp0156'),\n",
       " ('GenProp0142', 'GenProp0168'),\n",
       " ('GenProp0142', 'GenProp0231'),\n",
       " ('GenProp0142', 'GenProp0238'),\n",
       " ('GenProp0142', 'GenProp0239'),\n",
       " ('GenProp0142', 'GenProp0734'),\n",
       " ('GenProp0734', 'GenProp0139'),\n",
       " ('GenProp0017', 'GenProp0157'),\n",
       " ('GenProp0157', 'GenProp0055'),\n",
       " ('GenProp0157', 'GenProp0156'),\n",
       " ('GenProp0157', 'GenProp0168'),\n",
       " ('GenProp0017', 'GenProp0247'),\n",
       " ('GenProp0247', 'GenProp0075'),\n",
       " ('GenProp0075', 'GenProp0002'),\n",
       " ('GenProp0002', 'GenProp0791'),\n",
       " ('GenProp0791', 'GenProp0792'),\n",
       " ('GenProp0002', 'GenProp0723'),\n",
       " ('GenProp0075', 'GenProp0061'),\n",
       " ('GenProp0075', 'GenProp0127'),\n",
       " ('GenProp0075', 'GenProp2017'),\n",
       " ('GenProp0075', 'GenProp2021'),\n",
       " ('GenProp0247', 'GenProp0179'),\n",
       " ('GenProp0179', 'GenProp0052'),\n",
       " ('GenProp0179', 'GenProp0053'),\n",
       " ('GenProp0179', 'GenProp0059'),\n",
       " ('GenProp0179', 'GenProp0127'),\n",
       " ('GenProp0179', 'GenProp0207'),\n",
       " ('GenProp0179', 'GenProp0209'),\n",
       " ('GenProp0247', 'GenProp0197'),\n",
       " ('GenProp0197', 'GenProp0132'),\n",
       " ('GenProp0197', 'GenProp0188'),\n",
       " ('GenProp0197', 'GenProp0189'),\n",
       " ('GenProp0247', 'GenProp0248'),\n",
       " ('GenProp0248', 'GenProp0244'),\n",
       " ('GenProp0248', 'GenProp0245'),\n",
       " ('GenProp0248', 'GenProp0246'),\n",
       " ('GenProp0247', 'GenProp0301'),\n",
       " ('GenProp0301', 'GenProp2012'),\n",
       " ('GenProp0247', 'GenProp0302'),\n",
       " ('GenProp0302', 'GenProp0182'),\n",
       " ('GenProp0017', 'GenProp0303'),\n",
       " ('GenProp0303', 'GenProp0151'),\n",
       " ('GenProp0017', 'GenProp0731'),\n",
       " ('GenProp0731', 'GenProp0126'),\n",
       " ('GenProp0126', 'GenProp0001'),\n",
       " ('GenProp0126', 'GenProp0037'),\n",
       " ('GenProp0126', 'GenProp0109'),\n",
       " ('GenProp0126', 'GenProp0111'),\n",
       " ('GenProp0126', 'GenProp0117'),\n",
       " ('GenProp0126', 'GenProp0118'),\n",
       " ('GenProp0126', 'GenProp0159'),\n",
       " ('GenProp0126', 'GenProp0160'),\n",
       " ('GenProp0126', 'GenProp0162'),\n",
       " ('GenProp0126', 'GenProp0163'),\n",
       " ('GenProp0126', 'GenProp0164'),\n",
       " ('GenProp0126', 'GenProp0178'),\n",
       " ('GenProp0126', 'GenProp0199'),\n",
       " ('GenProp0199', 'GenProp0125'),\n",
       " ('GenProp0125', 'GenProp0160'),\n",
       " ('GenProp0125', 'GenProp0787'),\n",
       " ('GenProp0125', 'GenProp0786'),\n",
       " ('GenProp0199', 'GenProp0193'),\n",
       " ('GenProp0199', 'GenProp0788'),\n",
       " ('GenProp0788', 'GenProp0160'),\n",
       " ('GenProp0731', 'GenProp0733'),\n",
       " ('GenProp0733', 'GenProp0182'),\n",
       " ('GenProp0731', 'GenProp0734'),\n",
       " ('GenProp0734', 'GenProp0139'),\n",
       " ('GenProp0017', 'GenProp0820'),\n",
       " ('GenProp0820', 'GenProp0140'),\n",
       " ('GenProp0140', 'GenProp0137'),\n",
       " ('GenProp0140', 'GenProp0138'),\n",
       " ('GenProp0820', 'GenProp0770'),\n",
       " ('GenProp0017', 'GenProp0866'),\n",
       " ('GenProp0866', 'GenProp0865'),\n",
       " ('GenProp0065', 'GenProp0064'),\n",
       " ('GenProp0064', 'GenProp0501'),\n",
       " ('GenProp0064', 'GenProp0603'),\n",
       " ('GenProp0603', 'GenProp0029'),\n",
       " ('GenProp0029', 'GenProp0633'),\n",
       " ('GenProp0029', 'GenProp0631'),\n",
       " ('GenProp0029', 'GenProp0632'),\n",
       " ('GenProp0065', 'GenProp0066'),\n",
       " ('GenProp0066', 'GenProp0010'),\n",
       " ('GenProp0065', 'GenProp0067'),\n",
       " ('GenProp0065', 'GenProp0069'),\n",
       " ('GenProp0069', 'GenProp0297'),\n",
       " ('GenProp0065', 'GenProp0071'),\n",
       " ('GenProp0071', 'GenProp0179'),\n",
       " ('GenProp0179', 'GenProp0052'),\n",
       " ('GenProp0179', 'GenProp0053'),\n",
       " ('GenProp0179', 'GenProp0059'),\n",
       " ('GenProp0179', 'GenProp0127'),\n",
       " ('GenProp0179', 'GenProp0207'),\n",
       " ('GenProp0179', 'GenProp0209'),\n",
       " ('GenProp0071', 'GenProp0180'),\n",
       " ('GenProp0180', 'GenProp0119'),\n",
       " ('GenProp0180', 'GenProp0129'),\n",
       " ('GenProp0180', 'GenProp0172'),\n",
       " ('GenProp0180', 'GenProp0176'),\n",
       " ('GenProp0180', 'GenProp0190'),\n",
       " ('GenProp0180', 'GenProp0191'),\n",
       " ('GenProp0180', 'GenProp0192'),\n",
       " ('GenProp0180', 'GenProp0252'),\n",
       " ('GenProp0180', 'GenProp0873'),\n",
       " ('GenProp0065', 'GenProp0073'),\n",
       " ('GenProp0065', 'GenProp0074'),\n",
       " ('GenProp0074', 'GenProp0052'),\n",
       " ('GenProp0065', 'GenProp0076'),\n",
       " ('GenProp0076', 'GenProp0115'),\n",
       " ('GenProp0115', 'GenProp0114'),\n",
       " ('GenProp0115', 'GenProp0198'),\n",
       " ('GenProp0115', 'GenProp0216'),\n",
       " ('GenProp0115', 'GenProp0227'),\n",
       " ('GenProp0115', 'GenProp0228'),\n",
       " ('GenProp0076', 'GenProp0198'),\n",
       " ('GenProp0065', 'GenProp0167'),\n",
       " ('GenProp0167', 'GenProp0165'),\n",
       " ('GenProp0167', 'GenProp0166'),\n",
       " ('GenProp0167', 'GenProp0201'),\n",
       " ('GenProp0065', 'GenProp0475'),\n",
       " ('GenProp0065', 'GenProp0551'),\n",
       " ('GenProp0065', 'GenProp0864'),\n",
       " ('GenProp0864', 'GenProp0462'),\n",
       " ('GenProp0864', 'GenProp0874'),\n",
       " ('GenProp0065', 'GenProp1048'),\n",
       " ('GenProp0065', 'GenProp1067'),\n",
       " ('GenProp1067', 'GenProp0021')]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "parent_child_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('./gen_prop_mappings.tsv', 'w') as tsv_file:\n",
    "    csv_writer = csv.writer(tsv_file, delimiter='\\t')\n",
    "    for row in parent_child_table:\n",
    "        csv_writer.writerow(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gen_prop_path = '../genome-properties/docs/release/genomeProperties.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "from modules.lib import parse_genome_property_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "genome_property_flat_file_path = '../genome-properties/docs/release/genomeProperties.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(genome_property_flat_file_path) as genome_property_file:\n",
    "    properties = parse_genome_property_file(genome_property_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[GenProp0001, Type: PATHWAY, Name: Chorismate biosynthesis via shikimate, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0002, Type: GUILD, Name: Coenzyme F420 utilization, Thresh: 0, References: True, Databases: GUILD, Steps: True, Parent: True, Public: False,\n",
       " GenProp0010, Type: SYSTEM, Name: Inteins, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0017, Type: CATEGORY, Name: Metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0021, Type: GUILD, Name: CRISPR region, Thresh: 1, References: True, Databases: GUILD, Steps: True, Parent: False, Public: False,\n",
       " GenProp0023, Type: PATHWAY, Name: Glyoxylate shunt, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0029, Type: METAPATH, Name: Nitrogen fixation, Thresh: 0, References: True, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0030, Type: PATHWAY, Name: Glutathione biosynthesis, Thresh: 0, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0033, Type: METAPATH, Name: TCA cycle, Thresh: 5, References: False, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0036, Type: PATHWAY, Name: Biotin biosynthesis, Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0037, Type: PATHWAY, Name: Tryptophan biosynthesis from ribose-5-phosphate, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0038, Type: PATHWAY, Name: Tetrahydrofolate biosynthesis from GTP and PABA, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0046, Type: METAPATH, Name: IPP biosynthesis, Thresh: 0, References: False, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0047, Type: PATHWAY, Name: IPP biosynthesis via mevalonate, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0048, Type: PATHWAY, Name: IPP biosynthesis via deoxyxylulose, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0052, Type: SYSTEM, Name: Type III secretion, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0053, Type: SYSTEM, Name: Type II secretion, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0055, Type: SYSTEM, Name: Polyhydroxyalkanoic acids, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0057, Type: PATHWAY, Name: NAD(P) biosynthesis from L-aspartate and DHAP, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0058, Type: PATHWAY, Name: Menaquinone biosynthesis via SEPHCHC, Thresh: 3, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0059, Type: SYSTEM, Name: Type I secretion, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0061, Type: SYSTEM, Name: Lipoprotein system lgt/lsp/lnt, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0063, Type: CATEGORY, Name: Biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0064, Type: CATEGORY, Name: Biological niche, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0065, Type: CATEGORY, Name: Genome properties, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0066, Type: CATEGORY, Name: Selfish genetic elements, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0067, Type: CATEGORY, Name: Quantitative content, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0069, Type: CATEGORY, Name: Cell surface component, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0070, Type: CATEGORY, Name: Energy metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0071, Type: CATEGORY, Name: Transport, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0072, Type: CATEGORY, Name: Central intermediary metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0073, Type: CATEGORY, Name: Cell motility, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0074, Type: CATEGORY, Name: Virulence, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0075, Type: CATEGORY, Name: Protein modification, prosthetic groups and cofactors, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0076, Type: CATEGORY, Name: DNA handling, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0077, Type: CATEGORY, Name: Natural products biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0109, Type: PATHWAY, Name: Histidine biosynthesis from ribose-5-phosphate, Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0110, Type: PATHWAY, Name: Purine (inosine-5'-phosphate) biosynthesis from ribose-5-phosphate, Thresh: 4, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0111, Type: PATHWAY, Name: Proline biosynthesis from glutamate, Thresh: 1, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0114, Type: SYSTEM, Name: Nucleotide excision repair, Thresh: 0, References: False, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0115, Type: CATEGORY, Name: DNA repair, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0116, Type: CATEGORY, Name: DNA metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0117, Type: PATHWAY, Name: Arginine biosynthesis from ornithine, carbamoyl-p and aspartate, Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0118, Type: PATHWAY, Name: Ornithine biosynthesis from glutamate, acetylated branch, Thresh: 0, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0119, Type: SYSTEM, Name: PTS transport system, Thresh: 2, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0120, Type: PATHWAY, Name: Pentose phosphate cycle, Thresh: 4, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0124, Type: PATHWAY, Name: Pantothenate biosynthesis from aspartate and 2-oxoisovalerate, Thresh: 0, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0125, Type: METAPATH, Name: Lysine biosynthesis via diaminopimelate (DAP), Thresh: 1, References: True, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0126, Type: CATEGORY, Name: Amino acid biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0127, Type: SYSTEM, Name: Tat (Sec-independent) protein export, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0128, Type: SYSTEM, Name: F1/F0 ATPase, Thresh: 1, References: False, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0129, Type: SYSTEM, Name: Na+-translocating NADH-quinone reductase, Thresh: 2, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0130, Type: SYSTEM, Name: Rnf-type electron transport complex, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0131, Type: CATEGORY, Name: Electron transport, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0132, Type: SYSTEM, Name: Transcription termination/antitermination Nus factors, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0133, Type: CATEGORY, Name: Nucleic acid metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0136, Type: PATHWAY, Name: Ubiquinone biosynthesis from chorismate, aerobic, Thresh: 3, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0137, Type: SYSTEM, Name: Iron-sulfur cluster assembly SUF system, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0138, Type: SYSTEM, Name: Iron-sulfur cluster assembly iscSUA-hscBA-fdx system, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0139, Type: PATHWAY, Name: Degradation of tyrosine via homogentisate, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0140, Type: CATEGORY, Name: Iron-sulfur cluster assembly systems, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0141, Type: PATHWAY, Name: Galactose-6-phosphate degradation via tagatose-6-phosphate, Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0142, Type: CATEGORY, Name: Catabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0143, Type: PATHWAY, Name: Leloir pathway (galactose/glucose interconversion), Thresh: 0, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0144, Type: METAPATH, Name: Chlorophyllide a biosynthesis from protoporphyrin IX, Thresh: 1, References: True, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0146, Type: PATHWAY, Name: Bacteriochlorophyll biosynthesis from chlorophyllide a, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0147, Type: PATHWAY, Name: Glycine betaine biosynthesis from choline, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0150, Type: PATHWAY, Name: Chlorophyll biosynthesis from chlorophyllide a, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0151, Type: SYSTEM, Name: Resistance to mercury, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0155, Type: PATHWAY, Name: Dissimilatory sulfate reduction, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0156, Type: SYSTEM, Name: Cyanophycin-like storage polymers, Thresh: 0, References: False, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0157, Type: CATEGORY, Name: Storage polymer systems, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0159, Type: PATHWAY, Name: Threonine biosynthesis from aspartate semialdehyde, Thresh: 0, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0160, Type: PATHWAY, Name: Aspartate semialdehyde biosynthesis from aspartate, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0162, Type: PATHWAY, Name: Isoleucine biosynthesis from threonine and pyruvate, Thresh: 1, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0163, Type: PATHWAY, Name: Valine biosynthesis from pyruvate, Thresh: 1, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0164, Type: PATHWAY, Name: Leucine biosynthesis from pyruvate and acetyl-CoA, Thresh: 2, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0165, Type: SYSTEM, Name: MinCDE system, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0166, Type: SYSTEM, Name: MreBCD system, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0167, Type: CATEGORY, Name: Cellular growth, organization and division, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0168, Type: SYSTEM, Name: Glycogen system, Thresh: 1, References: False, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0170, Type: SYSTEM, Name: Coenzyme PQQ biosynthesis, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0171, Type: PATHWAY, Name: Coenzyme A biosynthesis from pantothenate, Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0172, Type: SYSTEM, Name: Potassium-transporting ATPase KdpFABC, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0176, Type: SYSTEM, Name: TRAP-T (tripartite ATP-independent periplasmic transporters) family transporters, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0178, Type: SYSTEM, Name: Glycine cleavage system, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0179, Type: CATEGORY, Name: Protein transport, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0180, Type: CATEGORY, Name: Small molecule transport, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0182, Type: GUILD, Name: Reduction of oxidized methionine, Thresh: 0, References: False, Databases: GUILD, Steps: True, Parent: False, Public: False,\n",
       " GenProp0183, Type: METAPATH, Name: dTDP-L-rhamnose biosynthesis from dTDP-4-dehydro-L-rhamnose, Thresh: 0, References: True, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0184, Type: CATEGORY, Name: Cofactor biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0185, Type: CATEGORY, Name: Nucleotide biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0186, Type: CATEGORY, Name: Storage and structural polymer biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0187, Type: PATHWAY, Name: Pyrimidine (uridine-5'-phosphate) de novo biosynthesis, Thresh: 4, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0188, Type: PATHWAY, Name: GatABC aspartyl/glutamyl-tRNA(Asn/Gln) amidotransferase complex, Thresh: 0, References: False, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0189, Type: PATHWAY, Name: tRNA-Gln direct aminoacylation, Thresh: 0, References: False, Databases: PATHWAY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0190, Type: SYSTEM, Name: Phosphate ABC transporter (pstSCAB-phoU), Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0191, Type: SYSTEM, Name: Sulfate/thiosulfate ABC transporter, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0192, Type: SYSTEM, Name: Molybdate ABC transporter, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0193, Type: PATHWAY, Name: Lysine biosynthesis via alpha-aminoadipate (AAA pathway), Thresh: 6, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0196, Type: CATEGORY, Name: Sulfur metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0197, Type: CATEGORY, Name: Protein biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0198, Type: SYSTEM, Name: RuvABC Holliday junction complex, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0199, Type: METAPATH, Name: Lysine biosynthesis, Thresh: 0, References: False, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0201, Type: SYSTEM, Name: SMC-ScpA-ScpB complex, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0202, Type: CATEGORY, Name: Nitrogen metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0207, Type: SYSTEM, Name: Lipoprotein localization system lolABCDE, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0209, Type: SYSTEM, Name: Sec system preprotein translocase, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0216, Type: SYSTEM, Name: RecBCD pathway, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0217, Type: PATHWAY, Name: 5-aminolevulinate biosynthesis (glutamate pathway), Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0223, Type: PATHWAY, Name: 5-aminolevulinate biosynthesis from succinyl-CoA and glycine, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0227, Type: CATEGORY, Name: Excision repair, Thresh: None, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0228, Type: CATEGORY, Name: Direct repair, Thresh: None, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0231, Type: PATHWAY, Name: 4-hydroxyphenylacetate degradation, Thresh: 3, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0238, Type: PATHWAY, Name: 2-aminoethylphosphonate catabolism to acetaldehyde, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0239, Type: CATEGORY, Name: Propionyl-CoA catabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0242, Type: CATEGORY, Name: Phosphorus metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0243, Type: CATEGORY, Name: ATP synthases, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0244, Type: SYSTEM, Name: Chaperone system: DnaK-DnaJ-GrpE, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0245, Type: SYSTEM, Name: Chaperone system: GroEL/GroES, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0246, Type: SYSTEM, Name: Chaperone system: thermosome/prefoldin, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0247, Type: CATEGORY, Name: Protein metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0248, Type: CATEGORY, Name: Protein folding, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0250, Type: PATHWAY, Name: Hydroxyethylthiazole (HET) biosynthesis from 1-deoxy-D-xylulose-5-phosphate, Cysteine, and either Tyr or Gly, Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0252, Type: SYSTEM, Name: Thiamine/TPP ABC transporter ThiBPQ, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0253, Type: PATHWAY, Name: Hydroxymethylpyrimidine pyrophosphate (HMP-P) biosynthesis from aminoimidazole ribotide, Thresh: 1, References: True, Databases: PATHWAY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0254, Type: PATHWAY, Name: Thiamine pyrophosphate (TPP) biosynthesis from HMP-PP and HET-P, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0255, Type: METAPATH, Name: 5-aminolevulinate biosynthesis, Thresh: 0, References: False, Databases: METAPATH, Steps: True, Parent: False, Public: False,\n",
       " GenProp0261, Type: METAPATH, Name: Thiamine pyrophosphate (TPP) de novo biosynthesis, Thresh: 0, References: True, Databases: METAPATH, Steps: True, Parent: False, Public: False,\n",
       " GenProp0262, Type: SYSTEM, Name: RNA polymerase, bacterial, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp0267, Type: CATEGORY, Name: Osmolyte biosynthesis, Thresh: 0, References: True, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0297, Type: CATEGORY, Name: Lipopolysaccharide biosynthesis, Thresh: 0, References: True, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0298, Type: CATEGORY, Name: Fermentation, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0299, Type: CATEGORY, Name: Fatty acid biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0300, Type: CATEGORY, Name: Photosynthetic cofactor biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0301, Type: CATEGORY, Name: Protein degradation, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0302, Type: CATEGORY, Name: Protein repair, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0303, Type: CATEGORY, Name: Detoxification, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0462, Type: CATEGORY, Name: Response to stress, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0475, Type: CATEGORY, Name: Unknown biological process, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0501, Type: CATEGORY, Name: Metrics of biological environment, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0551, Type: CATEGORY, Name: Regulatory systems, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0603, Type: CATEGORY, Name: Lifestyle, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0630, Type: SYSTEM, Name: 2-oxoglutarate dehydrogenase system, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0631, Type: SYSTEM, Name: Nitrogenase, Fe-only, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0632, Type: SYSTEM, Name: Nitrogenase, V-containing, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0633, Type: SYSTEM, Name: Nitrogenase, Mo-containing, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0723, Type: SYSTEM, Name: Hydrogenase, nickel and F420-dependent, Thresh: 0, References: True, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp0731, Type: CATEGORY, Name: Amino acid metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0733, Type: CATEGORY, Name: Amino acid salvage, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0734, Type: CATEGORY, Name: Amino acid catabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0770, Type: CATEGORY, Name: Siderophore biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0786, Type: PATHWAY, Name: Lysine biosynthesis via diaminopimelate (DAP), succinylated branch, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0787, Type: PATHWAY, Name: Lysine biosynthesis via diaminopimelate (DAP), acetylated branch, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp0788, Type: METAPATH, Name: Lysine biosynthesis via diaminopimelate (DAP) utilizing ammonia and NADPH, Thresh: 4, References: True, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp0791, Type: METAPATH, Name: Coenzyme F420 biosynthesis, Thresh: 1, References: True, Databases: METAPATH, Steps: True, Parent: False, Public: False,\n",
       " GenProp0792, Type: PATHWAY, Name: 7,8-didemethyl-8-hydroxy-5-deazariboflavin (FO) biosynthesis, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0795, Type: CATEGORY, Name: Carbohydrate biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0816, Type: CATEGORY, Name: Nucleoside catabolism/utilization pathways, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0817, Type: CATEGORY, Name: Nucleotide catabolism/utilization pathways, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0819, Type: CATEGORY, Name: RNA metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0820, Type: CATEGORY, Name: Iron metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0829, Type: PATHWAY, Name: Menaquinone biosynthesis via futalosine, Thresh: 2, References: True, Databases: PATHWAY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0836, Type: METAPATH, Name: Menaquinone biosynthesis, Thresh: 0, References: False, Databases: METAPATH, Steps: True, Parent: False, Public: False,\n",
       " GenProp0838, Type: CATEGORY, Name: Lipid biosynthesis, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0864, Type: CATEGORY, Name: Response to external environment, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0865, Type: CATEGORY, Name: Carbon fixation, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0866, Type: CATEGORY, Name: Carbon metabolism, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp0873, Type: CATEGORY, Name: Ion-motive antiporter cycles, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0874, Type: CATEGORY, Name: Response to acidic (low pH) environment, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp0950, Type: PATHWAY, Name: dTDP-4-dehydro-6-deoxy-alpha-D-glucose biosynthesis from glucose-1-phosphate, Thresh: 0, References: True, Databases: PATHWAY, Steps: True, Parent: True, Public: False,\n",
       " GenProp1004, Type: SYSTEM, Name: Light-independent protochlorophyllide reductase, Thresh: 0, References: False, Databases: SYSTEM, Steps: True, Parent: False, Public: False,\n",
       " GenProp1032, Type: METAPATH, Name: dTDP-4-dehydro-L-rhamnose biosynthesis from dTDP-4-dehydro-6-deoxy-alpha-D-glucose, Thresh: 0, References: True, Databases: METAPATH, Steps: True, Parent: True, Public: False,\n",
       " GenProp1048, Type: CATEGORY, Name: Quorum-sensing, biofilm formation, and developmental systems, Thresh: 0, References: False, Databases: CATEGORY, Steps: False, Parent: False, Public: False,\n",
       " GenProp1067, Type: CATEGORY, Name: Defense systems, Thresh: 0, References: False, Databases: CATEGORY, Steps: True, Parent: False, Public: False,\n",
       " GenProp2012, Type: SYSTEM, Name: 19S proteasome cap, Thresh: 2, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp2017, Type: SYSTEM, Name: Gamma secretase, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False,\n",
       " GenProp2021, Type: SYSTEM, Name: Apoptosome, Thresh: 1, References: True, Databases: SYSTEM, Steps: True, Parent: True, Public: False]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "properties"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "gen_prop_data_rows = []\n",
    "for genome_property in properties:\n",
    "    new_gen_prop_row = (genome_property.id, \n",
    "                        genome_property.name, \n",
    "                        genome_property.type, \n",
    "                        genome_property.description, \n",
    "                        genome_property.private_notes)\n",
    "    gen_prop_data_rows.append(new_gen_prop_row) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('GenProp0001',\n",
       "  'Chorismate biosynthesis via shikimate',\n",
       "  'PATHWAY',\n",
       "  'Chorismate is the final common intermediate in the biosynthesis of phenylalanine, tyrosine and tryptophan (the aromatic amino acids) as well as menaquinone, ubiquinone, salicylate and phenazine.Chorismate is synthesized via a seven-step pathway starting from the precursors D-erythrose 4-phosphate and phosphoenolpyruvate.This pathway is widely distributed among microorganisms[1]. Certain methanogenic euarchaeota appear to be missing the first two steps of the pathway - these may be catalyzed by as of yet uncharacterized enzymes.',\n",
       "  None),\n",
       " ('GenProp0002',\n",
       "  'Coenzyme F420 utilization',\n",
       "  'GUILD',\n",
       "  'Coenzyme F420 (a 7,8-didemethyl-8-hydroxy 5-deazaflavin) is found in methanogenic archaea, the Mycobacteria, and some other species[1]. Forms differ in the number of glutamyl residues attached. The chromophore by itself, FO, often can replace intact F420, so the literature will occasionally describe an enzyme dependent on FO. This does not necessarily mean the FO is not typically part of coenzyme F420 in vivo. However, it appears that only FO, and not F420, occurs in the Cyanobacteria. This property describes the presence of enzymes that use F420 as an electron donor or receptor, as opposed to enzymes of its biosynthesis.',\n",
       "  None),\n",
       " ('GenProp0010',\n",
       "  'Inteins',\n",
       "  'SYSTEM',\n",
       "  'Inteins are protein insertion sequences that are embedded in host protein sequences. They are post-translationally excised from the host protein by a self-catalytic protein splicing process, in which the intein sequence is precisely excised, and the flanking host protein sequences (N- and C-exteins) are re-ligated to create a functional protein. The region between the N- and C-terminal splicing regions may show endonuclease homology[1].',\n",
       "  None),\n",
       " ('GenProp0017',\n",
       "  'Metabolism',\n",
       "  'CATEGORY',\n",
       "  'The properties contained within this category describe the various metabolic pathways by which organisms transform chemicals in the processes of life.',\n",
       "  None),\n",
       " ('GenProp0021',\n",
       "  'CRISPR region',\n",
       "  'GUILD',\n",
       "  'CRISPR repeats are by definition Clustered Regularly Interspaced Short Palindromic Repeats. CRISPR loci are described as having direct repeats, varying in size from 21 to 37 bp, interspaced by similarly sized non-repetitive sequences. Certain proteins, including Cas1 to Cas4, are found only in genomes with CRISPR repeats, and always near these repeats [1]. Haft, et al. described subtypes in which certain groups of subtype-specific genes are found in addition to characteristic sets of the more widely distributed Cas genes [2]. A further review of CRISPR families by Makarova, et al. describes broader categories of CRISPR systems, using a polythetic classification system [3].',\n",
       "  None),\n",
       " ('GenProp0023',\n",
       "  'Glyoxylate shunt',\n",
       "  'PATHWAY',\n",
       "  'The glyxoylate shunt consists of two enzymes, isocitrate lyase and malate synthase. Its function is generally anaplerotic, meaning that it replenishes TCA cycle intermediates. Isocitrate (one intermediate) becomes succinate (one intermediate) plus glyoxylate. Glyoxylate plus an acetyl group from acetyl-CoA becomes malate, a second intermediate, for a gain of one. Acetyl groups, such as from fatty acid metabolism, by means of this pathway, can provide TCA cycle intermediates for use in amino acid biosynthesis and other biosynthetic pathways[1].',\n",
       "  None),\n",
       " ('GenProp0029',\n",
       "  'Nitrogen fixation',\n",
       "  'METAPATH',\n",
       "  'Nitrogen fixation is the conversion of atmospheric nitrogen, N2 (with its difficult-to-cleave triple bond) into usable forms such as NH3. Conditions should be anaerobic or microaerophilic. The characteristic enzyme is nitrogenase, but many other proteins are involved. Nitrogenase is a complex composed of dinitrogenase (MoFe protein), which is a homodimer, and dinitrogenase reductase (Fe protein) that is a heterotetramer. Expected along with nitrogenase are several accessory proteins, the transcriptional regulator NifA, a ferredoxin or flavodoxin, downstream genes for nitrogen assimilation (NtrC, GlnA), and genes for protection from oxygen (e.g. leghemoglobin in root nodule bacteria).',\n",
       "  None),\n",
       " ('GenProp0030',\n",
       "  'Glutathione biosynthesis',\n",
       "  'PATHWAY',\n",
       "  'Glutathione (GSH) is gamma-Glu-Cys-Gly, a thiol whose -SH group is an important reductant in many systems. Some species use thiols other than glutathione: mycothione in Mycobacterium tuberculosis, gamma-glutamylcysteine in Halobacterium, and trypanothione in trypanosomes. Cysteine and CoA may also serve. Glutathione biosynthesis is accomplished by two enzymes. First, glutamate--cysteine ligase links the gamma-COOH of Glu to the NH2 of Cys. Next, glutathione synthase creates a peptide bond from the COOH of Cys to the NH2 of Gly. Species that synthesize glutathione are likely to have a reductase for oxidized glutathione and a number of glutathione-dependent enzymes.',\n",
       "  None),\n",
       " ('GenProp0033',\n",
       "  'TCA cycle',\n",
       "  'METAPATH',\n",
       "  'The TCA cycle (also known as the Krebs cycle, citrate acid cycle, and tricarboxylic acid cycle) is performed in eukaryotic mitochondria and in many prokaryotes. This property refers to the forward direction: from oxaloacetate and acetate to citrate, to isocitrate, to 2-oxoglutarate, to succinate, to fumarate, to malate, to oxaloacetate, with the release of two molecules of CO2. It runs only under aerobic conditions. The state YES means the cycle is complete. A complete TCA cycle may correlate with use of TCA intermediates as biosynthetic precursors and with anaplerotic (TCA cycle intermediate-replenishing) reactions such as the glyoxylate shunt. The state PARTIAL indicates the pathway does not form a full cycle but may contain several consecutive steps. The direction of flux through this arc is not specified when the PARTIAL state is set, and so it may be in the reductive (reverse) direction. The full reverse TCA cycle, an alternative to the Calvin cycle for carbon dioxide fixation, is covered by a separate property.',\n",
       "  None),\n",
       " ('GenProp0036',\n",
       "  'Biotin biosynthesis',\n",
       "  'PATHWAY',\n",
       "  'Biotin is an essential cofactor for many carboxylation (addition of CO2) reactions. This property reflects biosynthesis of biotin from pimeloyl-CoA, although the source of pimeloyl-CoA may vary. 8-amino-7-oxononanoate synthase (also called 7-keto-8-aminopelargonic acid synthetase, AONS, gene BioF) converts pimeloyl-CoA to 8-amino-7-oxononanoate (AON). Adenosylmethionine-8-amino-7-oxononanoate aminotransferase (gene BioA) converts the product to 7,8-diaminononanoate, from which dethiobiotin synthase (gene BioD) makes dethiobiotin. Biotin synthase (gene BioB) then makes biotin itself[1]. Enzymes such as carboxylesterase (gene BioH), which are involved in pimeloyl-CoA biosynthesis typically receive biotin-related annotations, but may also appear in genomes in which biotin is not synthesized and pimeloyl-CoA is used for something else.',\n",
       "  None),\n",
       " ('GenProp0037',\n",
       "  'Tryptophan biosynthesis from ribose-5-phosphate',\n",
       "  'PATHWAY',\n",
       "  'The essential amino acid tryptophan is synthesized in prokaryotes, plants, and fungi via a conserved six step pathway from the branchpoint compound chrosimate, a common precursor in the biosynthesis of aromatic amino acids. Seven catalytic domains and two alpha/beta subunit complexes comprise this pathway. Individually expressed in some organisms, gene fusions encoding single proteins possessing two or more of these catalytic domains have been observed for every trp gene. Although the occurance of gene fusions have been shown to be phylogenetically unrelated, a conserved gene order can be discerned within the domain Bacteria. Occasionally gene(s) dissociations disrupt gene order as in Campylobacter jejuni and Deinococcus radiodurans, or the pathway is found split between two separate chromosomes as in Rhodobacter sphaeroides. In cyanobacteria, such as Nostoc and Anabaena, it is not uncommon to observe paralogous trp genes as part of a larger \"supraoperon\" including other aromatic amino acid synthesizing genes as well as their trp counterparts [1].',\n",
       "  None),\n",
       " ('GenProp0038',\n",
       "  'Tetrahydrofolate biosynthesis from GTP and PABA',\n",
       "  'PATHWAY',\n",
       "  'Polyglutamylated tetrahydrofolate is the central cofactor of one-carbon metabolism, and is biosynthesized from GTP, para-aminobenzoate (from chorismate), and glutamate in a six step reaction [1]. The first enzyme (GTP cyclohydrolase I) catalizes dihydroneopterin triphosphate from GTP in the biosynthesis of tetrahydrofolate in prokaryotes, of tetrahydrobiopterin in vertebrates, and of pteridine-containing pigments in insects. This pathway is carried out by 6 enzymes, the complete pathway being absent in animals.',\n",
       "  None),\n",
       " ('GenProp0046',\n",
       "  'IPP biosynthesis',\n",
       "  'METAPATH',\n",
       "  'Isopentenyl pyrophosphate (IPP) is the common intermediate of isoprenoid metabolism which includes the biosynthesis of terpenoid natural products, sterols, bile acids, carotenoids, vitamins K and E, isoprene and other prenylated compounds. IPP may be synthesized via one of two separate pathways, the mevalonate pathway is typical of eukaryotes while the deoxyxylulose pathway is typical of prokaryotes. This property summarizes the states of the two child properties, IPP biosynthesis via mevalonate and IPP biosynthesis from deoxyxylulose.',\n",
       "  None),\n",
       " ('GenProp0047',\n",
       "  'IPP biosynthesis via mevalonate',\n",
       "  'PATHWAY',\n",
       "  'Isoprenyl pyrophosphate (IPP) is synthesized in eukaryotes, archaea and certain bacteria by the mevalonate pathway (which is in contrast to the non-mevalonate pathway found more generally in prokaryotes).The bacteria containing this pathway are believed to have acquired it by lateral transfer, possibly in more than one independent event [1]. Archaea are believed to possess a mevalonate pathway for production of IPP, but several of the steps are divergent from the known enzymes in eukaryotes and have not yet been conclusively identified [2]. In eukaryotes this pathway is highly regulated to control the production of steroids which require IPP as a precursor. IPP is also required for the synthesis of menaquinone, ubiquinone, terpenoids, isoprene and prenylated proteins. The pathway begins with synthesis of (S)-3-hydroxy-3-methylglutaryl-CoA (HMG-CoA) from acetyl-CoA and acetoacetyl-CoA by HMG-CoA synthase. The pathway contains 5 steps all of which are included as required elements in this property. Some organisms contain a pathway-specific acetoacetyl-CoA synthase which is described in this property, but is not a required step. It is likely that other species obtain acetoacetyl-CoA from beta-oxidation of fatty acids or related processes.',\n",
       "  None),\n",
       " ('GenProp0048',\n",
       "  'IPP biosynthesis via deoxyxylulose',\n",
       "  'PATHWAY',\n",
       "  'Isoprenyl pyrophosphate (IPP) is synthesised in most bacteria by the non-mevalonate pathway (also known as the MEP pathway, the glyceraldehyde-3-phosphate(GAP)-pyruvate pathway and the deoxyxylulose pathway). This is in contrast to the mevalonate pathway found in eukaryotes. This non-mevalonate pathway contains six characterized enzymes and may involve additional proteins as well. IPP is a basic building block of isoprenoids, also known as terpenoids [1].',\n",
       "  None),\n",
       " ('GenProp0052',\n",
       "  'Type III secretion',\n",
       "  'SYSTEM',\n",
       "  'Type III secretion systems are found in a number of pathogenenic gram-negative bacteria. These systems consist of approximately twenty-five translocation proteins, as well as those proteins that are transported either across the bacterial inner and outer membrane, into the medium, or further across a eukaryotic target cell membrane. Substrates for type III secretion are not cleaved during transport, although the signal for transport appears to lie near the N-terminus (or in the RNA); chaperones act as well for some substrates and may be a second signal. Proteins delivered to the target cell cytosol and active in pathogenesis are termed effectors. Type III secretion systems resemble some of the flagellar export apparatus. In some cases, a type III pilus is observed and required for effector delivery. Gene nomeclature transposes across genomes: ysc in Yersinia, bsc in Bordetella, esc in enteropathogenic E. coli, hrc in plant pathogens, etc., so that YscD, EscD, HrcD, etc. are homologous and functionally analogous [1,2,3].',\n",
       "  None),\n",
       " ('GenProp0053',\n",
       "  'Type II secretion',\n",
       "  'SYSTEM',\n",
       "  'Type II secretion is also known as the main terminal branch of the general secretion pathway. Type II secretion systems transport folded proteins across the outer membrane. These proteins have previously come unfolded across the inner membrane by the general secretory pathway (Sec) and are then fully folded in the periplasm [1]. It has been reported that some proteins may cross the inner membrane by Tat (Sec-independent) protein export rather than by Sec prior to type II secretion [2].',\n",
       "  None),\n",
       " ('GenProp0055',\n",
       "  'Polyhydroxyalkanoic acids',\n",
       "  'SYSTEM',\n",
       "  'Polyhydroxyalkanoic acids (PHAs) are high molecular weight carbon and energy storage molecules. They exist as linear polymers in which the monomeric units have the form -[O-CHR(CH2)xCO]-. The aliphatic polyesters occur in inclusion bodies in various bacteria and archaea. The most common form is polyhydroxybutyric acid, -[0-CHCH3-CH2-CO]-. Typical biosynthetic enzymes include ketothiolases, reductases, and PHA synthases. Additional proteins may include small PHA inclusion body-associated proteins. Typical gene symbols include pha (phb for polyhydroxybutyrate) A (ketothiolase), B (reductase), C (synthase), Z (depolymerase), E (second subunit of heterodimeric-type PHA synthase), R, and Q. It has been reported that all prokaryotes and eukaryotes synthesize PHB of medium-chain length (~100200 residues) thats is found in lipid bilayers and vesicles, and PHB of short-chain length (<12 residues) which is conjugated to proteins in membranes and organelles [1]. However, this property refers to very high molecular weight (10,000 to >1,000,000 residues) PHAs in granular deposits.',\n",
       "  None),\n",
       " ('GenProp0057',\n",
       "  'NAD(P) biosynthesis from L-aspartate and DHAP',\n",
       "  'PATHWAY',\n",
       "  'Nicotinamide adenine dinucleotide (NAD+) and nicotinamide adenine dinucleotide phosphate (NADP+) also known as pyridine nucleotides, are cofactors involved in numerous redox reactions through reversible reactions to their reduced forms NADH and NADPH. In bacteria, NAD+ is synthesized in a 5 step reaction (NadABCDE) from L-aspartate, and NADP+ is made from NAD+ by the action of NAD kinase [1]. In mammals, an altenative pathway exists in which the intermediate quinolinate is made from L-tryptophan. NAD+ may also be recycled from nicotinamide after nucleosidease cleavage, or synthesized from imported nicotinate (niacin).',\n",
       "  None),\n",
       " ('GenProp0058',\n",
       "  'Menaquinone biosynthesis via SEPHCHC',\n",
       "  'PATHWAY',\n",
       "  'Menaquinone (also known as Vitamin K2) is one of the quinone cofactors which play an essential role in hydrogen transfer reactions, particularly during anaerobic respiration. Menaquinone in E. coli and various other bacteria is synthesized from chorismate, 2-oxoglutarate, polyprenyl-pyrophosphate and S-adenosylmethionine via SEPHCHC and SHCHC intermediates [1,2]. A very similar molecule (phylloquinone, vitamin K1) is found in cyanobacteria and plants which contains a shorter isoprenoid (phytyl) chain in which three of the four double bonds are reduced. In animals, both vitamins K1 and K2 act as cofactors for gamma-carboxylation of glutamate (Glu -> Gla) which is critical for coagulation factors. Note that an alternative pathway for menaquinone biosynthesis (not described by this property) operates in Helicobacter pylori, Campylobacter jejuni, Streptomyces, and other species [3].',\n",
       "  None),\n",
       " ('GenProp0059',\n",
       "  'Type I secretion',\n",
       "  'SYSTEM',\n",
       "  'A type I protein secretion system is an ABC (ATP-binding cassette) transporter. The system is Sec-independent. ATP drives the transport of target proteins across both inner and outer membranes and into the extracellular medium. There is no cleaved N-terminal signal sequence. Instead, the type I system recognizes C-terminal information, localized to the last 50 amino acids or less. Characteristic sequence found nearby, such as repeats of the motif GGXGXDXXX in Erwinia, may play a supporting role, possibly for chaperoning the rest of the protein rather than for recognition of the export signal [1].This property does NOT cover ABC transporter-mediated processing and secretion of class II bacteriocins of certain Gram-positive bacteria.',\n",
       "  None),\n",
       " ('GenProp0061',\n",
       "  'Lipoprotein system lgt/lsp/lnt',\n",
       "  'SYSTEM',\n",
       "  'Some membrane proteins are anchored to the membrane by a lipid moiety attached to a modified N-terminus rather than by a transmembrane helix. This property describes such a system. The system starts with the modification of the Cys sulfhydryl of the conserved lipobox motif L-X-G/A-C by prolipoprotein diacylglyceryl transferase (Lgt). Prolipoprotein signal peptidase (Lsp) then cleaves the lipoprotein signal sequence leaving Cys as the new N-terminal residue. Finally the Cys NH2 is acylated by apolipoprotein N-acyltransferase (Lnt) [1].',\n",
       "  None),\n",
       " ('GenProp0063',\n",
       "  'Biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The process of creating complex biomolecules from simpler starting materials.',\n",
       "  None),\n",
       " ('GenProp0064',\n",
       "  'Biological niche',\n",
       "  'CATEGORY',\n",
       "  'Properties which describe the typical or optimal environment for the growth of the organism. This excludes places or contexts where biologically inert forms (such as spores) of an organism may be found.',\n",
       "  None),\n",
       " ('GenProp0065', 'Genome properties', 'CATEGORY', None, None),\n",
       " ('GenProp0066',\n",
       "  'Selfish genetic elements',\n",
       "  'CATEGORY',\n",
       "  'Elements that enhance their own transmission relative to the rest of an individuals genome, but are neutral or harmful to the individual as a whole.',\n",
       "  None),\n",
       " ('GenProp0067', 'Quantitative content', 'CATEGORY', None, None),\n",
       " ('GenProp0069',\n",
       "  'Cell surface component',\n",
       "  'CATEGORY',\n",
       "  'Structures or systems which are localized at the surface of the cell and in contact with the external environment.',\n",
       "  None),\n",
       " ('GenProp0070',\n",
       "  'Energy metabolism',\n",
       "  'CATEGORY',\n",
       "  'Processes by which chemical or photonic energy is converted into a common currency which is used to power the processes of life.',\n",
       "  None),\n",
       " ('GenProp0071',\n",
       "  'Transport',\n",
       "  'CATEGORY',\n",
       "  'Process by which molecules are moved across biological membranes.',\n",
       "  None),\n",
       " ('GenProp0072', 'Central intermediary metabolism', 'CATEGORY', None, None),\n",
       " ('GenProp0073', 'Cell motility', 'CATEGORY', None, None),\n",
       " ('GenProp0074',\n",
       "  'Virulence',\n",
       "  'CATEGORY',\n",
       "  'Properties related to the ability of organisms to cause illness or damage to host species.',\n",
       "  None),\n",
       " ('GenProp0075',\n",
       "  'Protein modification, prosthetic groups and cofactors',\n",
       "  'CATEGORY',\n",
       "  'Properties related to those chemical species which become covalently bound to, or otherwise interact with proteins in ways that alter their biological function.',\n",
       "  None),\n",
       " ('GenProp0076',\n",
       "  'DNA handling',\n",
       "  'CATEGORY',\n",
       "  'Properties related to the utilization, maintenance, duplication and exchange of DNA.',\n",
       "  None),\n",
       " ('GenProp0077',\n",
       "  'Natural products biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The biosynthesis of complex biomolecules generally as polyketides, non-ribosomal peptides or polyisoprenoids (terpenoids). Often these molecules are extensively modified by methylation, oxidation/reduction, cyclization, epimerization and rearrangement reactions. Common functions for such compounds are as siderophores and toxins (antibiotics). Most often these compounds are exported from the cell.',\n",
       "  None),\n",
       " ('GenProp0109',\n",
       "  'Histidine biosynthesis from ribose-5-phosphate',\n",
       "  'PATHWAY',\n",
       "  \"Histidine is synthesized from phosphoribosyl-pyrophosphate in a ten step pathway found in bacteria, archaebacteria, fungi, and plants. The early stage of this pathway takes a ribose and transforms it into imadiazoleglycerol phosphate. Once the imadiazole ring is formed, glutamate donates the a-amino group and the newly formed amine is oxidized to histidine [1]. In Salmonella typhimurium and Escherichia coli, this pathway is carried out by the eight genes in the 'his' operon. hisI, hisB, and hisD are bifunctional enzymes, hisF and hisH form a monofunctional dimer, and the rest are monofunctional enzymes [2].\",\n",
       "  None),\n",
       " ('GenProp0110',\n",
       "  \"Purine (inosine-5'-phosphate) biosynthesis from ribose-5-phosphate\",\n",
       "  'PATHWAY',\n",
       "  \"Inosine-5-phosphate (IMP) is synthesized in a ten step pathway from phosphoribosyl- pyrophosphate (PRPP) and is the precursor of all purine nucleosides (adenosine,guanosine, inosine and xanthosine). The purine ring is built from glutamine (2 nitrogens), glycine (2 carbons and a nitrogen), 10-formyl-THF (2 carbons), carbondioxide (1 carbon) and aspartate (1 nitrogen). An intermediate, AICAR-P, is formed during the process of histidine biosynthesis in a 1:1 ratio with histidine, but this route is generally insufficient for fulfilling the purine needs of a dividing cell and is not considered 'de novo' purine biosynthesis.\",\n",
       "  None),\n",
       " ('GenProp0111',\n",
       "  'Proline biosynthesis from glutamate',\n",
       "  'PATHWAY',\n",
       "  'The amino acid proline is synthesized in three steps from glutamate via L-glutamate-5-semialdehyde which non-enzymatically cyclizes to delta-1-pyrroline-5-carboxylate. Glutamate semialdehyde can also arise from transamination of ornithine. Proline may also be synthesized directly from ornithine by the action of ornithine cyclodeaminase. These alternate paths are not covered by this property.',\n",
       "  None),\n",
       " ('GenProp0114',\n",
       "  'Nucleotide excision repair',\n",
       "  'SYSTEM',\n",
       "  'One mechanism of DNA repair is nucleotide excision repair. Single-stranded cuts are made on both sides of a DNA lesion and the oligomer between the cuts is removed. Very few prokaryotic species lack this mechanism, although the insect endosymbiont Buchnera aphidicola is an example. In bacteria, the excinuclease UvrABC performs only this function, so the presence of any one of the three subunits suggests the mechanism is (or was) present. Also required is UvrD or a related helicase with similar effect, but this helicase acts in other processes as well. The status of nucleotide excision repair in the archaea is not fully defined. Eukaryotic NER differs in that many more genes are involved, and the characteristic size of the oligomer removed is different.',\n",
       "  None),\n",
       " ('GenProp0115',\n",
       "  'DNA repair',\n",
       "  'CATEGORY',\n",
       "  'Any of a number of mechanisms by which damage to DNA is repaired.',\n",
       "  None),\n",
       " ('GenProp0116',\n",
       "  'DNA metabolism',\n",
       "  'CATEGORY',\n",
       "  'The chemical reactions and physical changes involving DNA (deoxyribonucleic acid), one of the two main types of nucleic acid, consisting of a long, unbranched macromolecule formed from one, or more commonly two, strands of linked deoxyribonucleotides.',\n",
       "  None),\n",
       " ('GenProp0117',\n",
       "  'Arginine biosynthesis from ornithine, carbamoyl-p and aspartate',\n",
       "  'PATHWAY',\n",
       "  'The arginine biosynthesis pathway is a three step process and a part of the urea cycle. The first enzyme, ornithine carbamoyltransferase (OTCase) carries out the reaction anabolically in arginine biosynthesis, but in some cases, it carries out catabolic reactions. Most OTCases are homotrimers, but the homotrimers are organized into dodecamers built from four trimers in at least two species; the catabolic OTCase of Pseudomonas aeruginosa is allosterically regulated, while OTCase of the extreme thermophile Pyrococcus furiosus shows both allostery and thermophily [1]. The third step of the pathway yields L-arginine by argininosuccinate lyase, and the amino acid can be cleaved by arginase, yielding urea and reconstituting ornithine.',\n",
       "  None),\n",
       " ('GenProp0118',\n",
       "  'Ornithine biosynthesis from glutamate, acetylated branch',\n",
       "  'PATHWAY',\n",
       "  'This pathway results in the creation of ornithine for use in the biosynthesis of arginine and in the urea cycle. Ornithine can also be interconverted with glutamate semialdehyde from the proline biosynthetic pathway. This pathway represents one of two possible branches for the creation of ornithine from glutamate.',\n",
       "  None),\n",
       " ('GenProp0119',\n",
       "  'PTS transport system',\n",
       "  'SYSTEM',\n",
       "  'Phosphotransferase system (PTS) is a phosphorylation cascade that allows various carbohydrates to be transported accross a membrane at the expense of phosphoenolpyruvate (PEP). First, a phosphoryl group is donated from PEP to a general (non sugar-specific) enzyme I (phosphotransferase system, enzyme I), and then on to a phosphocarrier protein (HPr). HPr then dontaes the phosphoryl group to the sugar-specific EIIA (enzyme IIA), and then on to EIIB (enzyme IIB), before being transferred to the incoming carbohydrate. The carbohydrate is transported by EIIC (enzyme IIC), and occasionally EIID (enzyme IID), which are sugar-specific transmembrane components [1].',\n",
       "  None),\n",
       " ('GenProp0120',\n",
       "  'Pentose phosphate cycle',\n",
       "  'PATHWAY',\n",
       "  'The pentose phosphate cycle (also known as the hexose monophosphate shunt) is a metabolic pathway that generates NADPH when glucose 6-phosphate is oxidised to ribose 5-phosphate. Ribose 5-phosphate is a precursor for biosynthesis of nucleotides [1]. NADPH is used in many syntheses, including fatty acid synthesis.',\n",
       "  None),\n",
       " ('GenProp0124',\n",
       "  'Pantothenate biosynthesis from aspartate and 2-oxoisovalerate',\n",
       "  'PATHWAY',\n",
       "  'This genome property tracks four proteins of pantothenate biosynthesis from aspartate; PanB, PanC, PanD and PanE. In the simplest case PanB (3-methyl-2-oxobutanoate hydroxymethyltransferase) makes 2-dehydropantoate from 3-methyl-2-oxobutanoate. PanE (2-dehydropantoate 2-reductase) makes (R)-pantoate from 2-dehydropantoate. PanD (aspartate 1-decarboxylase) makes beta-alanine from L-aspartate, and PanC (pantoate--beta-alanine ligase) then produces (R)-pantothenate from beta-alanine and (R)-pantoate. The genes for three of the proteins (PanB, PanC and PanD) tend to be clustered in bacterial genomes. However, the protein family distribution for PanE, is very different. The production of (R)-pantoate from 2-dehydropantoate can occur in more than one way (NADH or NADPH-dependent), and may be catalyzed by enzymes that have additional important substrates (an example is the D-mandelate dehydrogenase of Enterococcus faecalis).',\n",
       "  None),\n",
       " ('GenProp0125',\n",
       "  'Lysine biosynthesis via diaminopimelate (DAP)',\n",
       "  'METAPATH',\n",
       "  'The basic amino acid lysine is synthesized in most bacteria and plants via a nine step pathway from aspartate (via aspartate semialdehyde which is also used in a number of other pathways). Most species utilize succinate (from succinyl-CoA) to make an important amide linkage in the ring-opening of piperidine dicarboxylate. The succinate is then hydrolyzed several steps later. Other species carry out analogous transformations with acetate (from acetyl-CoA). Chlamydia and cyanobacteria have been shown to utilize a direct aminotransferase path without the acylation/deacylation steps, and a number of other species apparently also use this variant [1]. Each of these variants are included within this property. Certain obligate intracellular organisms such as Coxiella, Rickettsia and Wolbachia, contain all but the final step (diaminopimelate decarboxylase) of this pathway. Presumably these organisms have no need of lysine biosynthesis, being able to obtain it from their hosts. The products of the penultimate two steps however, LL- and meso-diaminopimelate, are required components of the bacterial cell wall, and so the rest of the pathway persists. There is a possibility that the decarboxylase exists in these organsims but is undetected.',\n",
       "  None),\n",
       " ('GenProp0126',\n",
       "  'Amino acid biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The synthesis of the basic twenty amino acids found in proteins.',\n",
       "  None),\n",
       " ('GenProp0127',\n",
       "  'Tat (Sec-independent) protein export',\n",
       "  'SYSTEM',\n",
       "  'The twin-arginine dependent translocation (Tat) system of protein transport differs from Sec-dependent translocation in that the Tat system transports folded proteins while Sec feeds through unfolded, nascent proteins that fold after they cross the membrane. The significance is that Tat lets cofactors such as flavins or iron-sulfur clusters be retained during transit across the membrane. Targets for the system can often be recognized by a characteristic signal sequence with an Arg-Arg motif. The target proteins usually are predicted to perform redox functions [1]. This property reflects detection of the whole system: components of the translocase and at least one example of a predicted Tat signal sequence. Every system must have at least one TatA homolog (other than TatB) and at least one TatC homolog. TatB occurs only in Proteobacteria and is required at least in E. coli. Recently, several proteins (TorD, NapD) have been shown to act as chaperones that bind the TAT signal peptide of a specific protein and delay translocation until the cofactor has bound [2].',\n",
       "  None),\n",
       " ('GenProp0128',\n",
       "  'F1/F0 ATPase',\n",
       "  'SYSTEM',\n",
       "  'F1/F0 ATP synthase: F1 and F0 represent two major clusters of subunits. F1 is the peripheral membrane associated cluster. The F1 subunits: alpha, beta, gamma, delta, epsilon, have the same names both in prokaryotes and eukaryotes. F0 is the integral membrane cluster. In F0, nomenclatures differ. For example, prokaryotic chain A matches eukaryotic chain 6.',\n",
       "  None),\n",
       " ('GenProp0129',\n",
       "  'Na+-translocating NADH-quinone reductase',\n",
       "  'SYSTEM',\n",
       "  'Na+-translocating NADH:ubiquinone oxidoreductase is found in marine, halophilic, and pathogenic Gram-negative bacteria. The NQR complex contains six structural subunits and acts as Na(+) pump [1].',\n",
       "  None),\n",
       " ('GenProp0130',\n",
       "  'Rnf-type electron transport complex',\n",
       "  'SYSTEM',\n",
       "  'The six subunit complex RnfABCDGE in Rhodobacter capsulatus encodes an apparent NADH oxidoreductase responsible for electron transport to nitrogenase, necessary for nitrogen fixation [1,2]. A closely related complex in E. coli, RsxABCDGE (Reducer of SoxR), reduces the 2Fe-2S-containing superoxide sensor SoxR, active as a transcription factor when oxidized [3]. This family of putative NADH oxidoreductase complexes exists in many of the same species as the related NQR, a Na(+)-translocating NADH-quinone reductase, but is distinct.',\n",
       "  None),\n",
       " ('GenProp0131',\n",
       "  'Electron transport',\n",
       "  'CATEGORY',\n",
       "  'The transport of electrons from an electron donor to an electron acceptor.',\n",
       "  None),\n",
       " ('GenProp0132',\n",
       "  'Transcription termination/antitermination Nus factors',\n",
       "  'SYSTEM',\n",
       "  'This Genome Property describes the set of Nus factors NusA, NusB, and NusG, as well as ribosomal protein S10 which is known as NusE. The name Nus, for N-utilization substance, reflects the phenomenology of exploitation of N factors for phage lambda in Escherichia coli. Much of the literature of these transcription termination/anti-termination factors revolves around the interaction with the transcription termination factor Rho, but factor Rho and cis-acting Rho-utilization sites are less well distributed than the N-factors, which are essentially universal in bacteria. In species with Rho, Nus factors improve transcriptase turnover speeds. Nus factors can block Rho action, a process called anti-termination [1]. See GenProp0812 for Rho-dependent transcription termination.',\n",
       "  None),\n",
       " ('GenProp0133', 'Nucleic acid metabolism', 'CATEGORY', None, None),\n",
       " ('GenProp0136',\n",
       "  'Ubiquinone biosynthesis from chorismate, aerobic',\n",
       "  'PATHWAY',\n",
       "  'Ubiquinone (also known as coenzyme Q) is an electron and hydrogen transport cofactor important for aerobic respiration and photosynthesis. It is synthesized in bacteria from chorismate and polyprenyl pyrophosphates of varying lengths. The pathway is composed of 10 Ubi proteins which are named UbiA-I and UbiX. UbiG catalyses two O-methylation steps within the pathway, but with different substrates in each case [1]. UbiB is thought to be involved with the fourth step in the pathway, catalysed by UbiI [2], but is not included in the pathway diagrams. Knock-out of the three hydroxylation steps requiring molecular oxygen results in anaerobic synthesis of ubiquinone, indicating the presence of an alternative system [3]. In yeast and animals, a different pathway exists starting from 4-hydroxyphenylpyruvate, having latter steps in common with the bacterial pathway [4]. In addition to ubiquinone, bacteria may contain menaquinone and/or phylloquinone which together make up their quinone pool.',\n",
       "  None),\n",
       " ('GenProp0137',\n",
       "  'Iron-sulfur cluster assembly SUF system',\n",
       "  'SYSTEM',\n",
       "  'The iron-sulfur cluster assembly SUF system is one of three iron-sulfur assembly systems present in a wide range of bacteria, archeae, and plastids. The other two mechanisms for iron-sulfur cluster assembly are the ISC (iron-sulfur cluster) system, composed of at least six components and found broadly distributed from bacteria to higher eukaryotes, and the NIF (nitrogen fixation) system, which is required for nitrogenase maturation [1]. The SUF (mobilization of sulfur) system is composed of a six gene cluster with protein complexes responsible for cysteine desulfuration of sulfur atoms from cysteine, cytosolic ATPase activity, and a protein scaffold for iron and sulfur cluster assembly and transfer to protein targets [2]. Operational under stressful growth conditions, such as iron limitation and oxidative stress, the SUF system has been shown to be an important factor for bacterial pathogenicity [3].',\n",
       "  None),\n",
       " ('GenProp0138',\n",
       "  'Iron-sulfur cluster assembly iscSUA-hscBA-fdx system',\n",
       "  'SYSTEM',\n",
       "  'At least three different systems have been described for 2Fe-2S cluster assembly. The NIF system provides an iron-sulfur cluster to nitrogenase in certain nitrogen-fixing species. SUF, described the most recently, can operate despite aerobic stress. This property describes the ISC system as found in E. coli and many other proteobacteria, containing at least the following elements: a characteristic cysteine desulfurase IscS, a characteristic ferredoxin, chaperones called heat shock cognate proteins A and B, and an assembly scaffold protein resembling the N terminus of the scaffold protein of the NIF system [1].',\n",
       "  None),\n",
       " ('GenProp0139',\n",
       "  'Degradation of tyrosine via homogentisate',\n",
       "  'PATHWAY',\n",
       "  'Tyrosine is degraded to produce fumarate and acetoacetate via the intermediate homogentisate, in a five-step pathway. Tyrosine is converted to 4-hydroxyphenylpyruvate by TyrB (tyrosine transaminase also known as tyrosine aminotransferase or PhhC) and then to homogentisate by Hpd (4-hydroxyphenylpyruvate dioxygenase). The second part of the pathway (the catabolism of homogentisate) involves three enzymes: HmgA (homogentisate dioxygenase); HmgB (fumarylacetoacetate hydrolase, also known as Fah); and HmgC (maleylacetoacetate isomerase, also known as Mai) [1].',\n",
       "  None),\n",
       " ('GenProp0140',\n",
       "  'Iron-sulfur cluster assembly systems',\n",
       "  'CATEGORY',\n",
       "  'At least three different systems have been described for iron cluster assembly, the NIF system provides clusters to nitrogenase, the ISC system may provide both 2Fe-2S and 4Fe-4S clusters for various enzymes in Proteobacteria and the SUF system is capable of functioning in the presence of oxidative stress.',\n",
       "  None),\n",
       " ('GenProp0141',\n",
       "  'Galactose-6-phosphate degradation via tagatose-6-phosphate',\n",
       "  'PATHWAY',\n",
       "  'Galactose-6-phosphate is degraded via the tagatose-6-phosphate pathway, which is a 3 step pathway encoded by the lacABCD gene cluster of the lactose-PTS (lactose phosphotransferase) operon. The three enzymes in the pathway are galactose-6-phosphate isomerase (this requires both LacA and LacB subunits for activity), tagatose-6-phosphate kinase (LacC), and tagatose 1,6-diphosphate aldolase (LacD)[1].',\n",
       "  None),\n",
       " ('GenProp0142',\n",
       "  'Catabolism',\n",
       "  'CATEGORY',\n",
       "  'Any metabolic process involving the breakdown of complex substances into smaller products, including the breakdown of carbon compounds with the liberation of energy for use by the cell or organism.',\n",
       "  None),\n",
       " ('GenProp0143',\n",
       "  'Leloir pathway (galactose/glucose interconversion)',\n",
       "  'PATHWAY',\n",
       "  'The Leloir pathway can interconvert galactose and glucose, often for galactose degradation. It tends to be found in the Firmicutes. It includes aldose 1-epimerase, galactokinase, galactose-1-P uridylyltransferase, and UDP-galactose 4-epimerase.',\n",
       "  None),\n",
       " ('GenProp0144',\n",
       "  'Chlorophyllide a biosynthesis from protoporphyrin IX',\n",
       "  'METAPATH',\n",
       "  'Chlorophyllide a is synthesised from protoporphyrin IX as part of the chlorophyll biosynthesis pathway. There is an aerobic and an anaerobic branch, this property represnts the aerobic branch. The first step (Mg-chelatase) has been shown to require three subunits encoded by three different genes. The final two steps in this pathway (vinyl reductase and protochlorophyllide reductase) can occur in the alternate order from that shown. There are two different enzymes that can catalyse the final (protochlorophyllide reductase) step; light-dependent protochlorophyllide reductase, and light-independent protochlorophyllide reductase. Despite catalyzing the same reaction, the two enzymes are very different [1].',\n",
       "  None),\n",
       " ('GenProp0146',\n",
       "  'Bacteriochlorophyll biosynthesis from chlorophyllide a',\n",
       "  'PATHWAY',\n",
       "  'Chlorophyllide a is the common branchpoint of the chlorophyll and bacteriochlorophyll biosynthetic pathways. This Genome Property describes the pathway beyond this branchpoint, which ultimately produces bacteriochlorophyll a. Three of the steps in the pathway (chlorophyllide reductase, bacteriochlorophyllide a dehydrogenase and bacteriochlorophyllide hydratase) can act in multiple orders producing different intermediates [1].',\n",
       "  None),\n",
       " ('GenProp0147',\n",
       "  'Glycine betaine biosynthesis from choline',\n",
       "  'PATHWAY',\n",
       "  'As a coping mechanism against osmotic stress, prokaryotic organisms accumulate osmolytes or osmoprotectants from exogenous sources, to counterbalance elevated osmolarities in the environment without affecting overall cell metabolism and structural integrity. One of the most efficient osmoprotectants found in the environment is the compound glycine betaine [1]. Both gram positive and gram negative bacteria mediate osmotolerance either by direct uptake of glycine betaine from the environment, or by synthesis from exogenously supplied choline [2,3]. This genome property refers to the biosynthesis of the osmoprotectant glycine betaine from choline. Biosynthesis of glycine betaine occurs via a two-step oxidative reaction requiring a terminal electron acceptor. Choline dehydrogenase first catalyzes the conversion of exogenously supplied choline into the intermediate glycine betaine aldehyde, with subsequent oxidization by betaine aldehyde dehydrogenase into the osmoprotectant glycine betaine. Both choline dehydrogenase and betaine aldehyde dehydrogenase are required for conversion of choline into glycine betaine [2,3].',\n",
       "  None),\n",
       " ('GenProp0150',\n",
       "  'Chlorophyll biosynthesis from chlorophyllide a',\n",
       "  'PATHWAY',\n",
       "  'Chlorophyll is formed directly from chlorophyllide a by the addition of a phytyl side chain, or by addition of a geranylgeranyl side chain and subsequent reduction of the side chain.The synthetase and reductase responsible for this conversion are closely related to the enzymes which carry out the analagous steps of bacteriochlorophyll biosynthesis [1].',\n",
       "  None),\n",
       " ('GenProp0151',\n",
       "  'Resistance to mercury',\n",
       "  'SYSTEM',\n",
       "  'Detoxification of inorganic and organic mercury compounds involves the enzymatic reduction of extracellular mercuric ions or organomercurials into less-toxic metallic mercury, which is released into the cytoplasm and then volatilizes from the cell. Mercuric ions diffuse across the outer membrane by a mer transport system, which binds and transfers mercury ions via a pair of cysteine residues to the active site of a mercuric reductase. Alternatively, an organomercurial lyase is required for hydrolysis of the carbon-mercury bond of organomercurial compounds before reduction proceeds. Transcribed separately and divergently, a mercury-sensitive transcriptional activator both positively and negatively regulates mer expression, while a co-transcribed secondary regulatory protein down-regulates this mer operon. This resistance system, often encoded on plasmids and transferred via transposons, is found in both gram-positive and gram-negative bacteria [1].',\n",
       "  None),\n",
       " ('GenProp0155',\n",
       "  'Dissimilatory sulfate reduction',\n",
       "  'PATHWAY',\n",
       "  'During anaerobic respiration, sulfate reducing bacteria catalyze the six electron reduction of sulfate to hydrogen sulfide. These sulfate reducers comprise both gram-positive and gram-negative bacteria, as well as archeae [1]. Sulfur oxidizing bacterium, such as Chromatium vinosum, can reversibly catalyze this pathway freeing electrons and ATP.',\n",
       "  None),\n",
       " ('GenProp0156',\n",
       "  'Cyanophycin-like storage polymers',\n",
       "  'SYSTEM',\n",
       "  'Cyanophycin is a characteristically Cyanobacterial storage polymer for nitrogen, carbon, and energy. It is made by cyanophycin synthetase and degraded by cyanophycinase. Cyanophycinase is also found in an extracellular form by some species that can utilize exogenous cyanophycin. The polymer has a backbone of L-aspartic acid residues (Asp), with the side chain COOH linked to the NH2 of arginine (Arg) residues. The native cyanobacterial form is insoluble, while heterogenously expressed cyanophycin synthetases in E. coli may produce a soluble form with some level of Lys substituting for Arg.This polymer is analogous to polyhydroxyalkanoic acid biosynthesis systems with which it sometimes co-occurs.',\n",
       "  None),\n",
       " ('GenProp0157',\n",
       "  'Storage polymer systems',\n",
       "  'CATEGORY',\n",
       "  'These polymers of various monomer units are used as storage for energy, carbon and/or nitrogen. Sometimes these polymers are observed as granular inclusions within the cell. These systems may include enzymes for the biosynthesis and degradation of the polymer, regulatory components as well as proteins associated with the granule surface.',\n",
       "  None),\n",
       " ('GenProp0159',\n",
       "  'Threonine biosynthesis from aspartate semialdehyde',\n",
       "  'PATHWAY',\n",
       "  'Threonine is synthesised from aspartate semialdehyde, a branch point intermediate in the synthesis of lysine. The aspartate semialdehyde is derived from aspartate, which in turn is derived from the Krebs cycle intermediate oxaloacetate. Homoserine, an intermediate in this pathway is a branchpoint for the biosynthesis of methionine. Threonine itself is utilized as a starting point for the biosynthesis of isoleucine.',\n",
       "  None),\n",
       " ('GenProp0160',\n",
       "  'Aspartate semialdehyde biosynthesis from aspartate',\n",
       "  'PATHWAY',\n",
       "  'Aspartate semialdehyde biosynthesis is a two step pathway. The product of this pathway can be used in lysine biosynthesis, or in the pathway leading to homoserine, which is involved in threonine, isoleucine and methionine biosynthesis. There are three isozymes that differ in their sensitivity to regulation by Lys, Met and Thr in Escherichia coli [1].',\n",
       "  None),\n",
       " ('GenProp0162',\n",
       "  'Isoleucine biosynthesis from threonine and pyruvate',\n",
       "  'PATHWAY',\n",
       "  'The branched amino acid isoleucine is biosynthesized from the central intermediary metabolite pyruvate and the amino acid threonine via a five step pathway. Four enzymes in this pathway are common to the pathways for the biosynthesis of leucine and valine, although specialized versions of these enzymes may exist in certain species which have a narrowed substrate range or amino acid-specific regulatory mechanisms. Unlike those other two pathways in which two molecules of pyruvate are condensed, here threonine is converted first to 2-oxobutyrate and then condensed with pyruvate. Thus, the enzymes common to the pathways are acting on distinct substrates. 2-oxobutyrate for this pathway may also be generated by a separate path from glutamate via mesaconate in certain organisms. The penultimate intermediate in this pathway, 2-oxo-3-methylvalerate may also be incorporated into branched-chain fatty acids',\n",
       "  None),\n",
       " ('GenProp0163',\n",
       "  'Valine biosynthesis from pyruvate',\n",
       "  'PATHWAY',\n",
       "  'The branched amino acid valine is biosynthesized from the central intermediary metabolite pyruvate via a four step pathway. All four enzymes in this pathway are common to the pathways for the biosynthesis of isoleucine from 2-oxobutyrate and leucine from pyruvate, although specialized versions of these enzymes may exist in certain species which have a narrowed substrate range or amino acid-specific regulatory mechanisms. The product of the third step in this pathway, 2-oxoisovalerate is a branchpoint metabolite required for the biosynthesis of Coenzyme A, branched-chain fatty acids, and leucine. Similarly, the product of the first step, (S)-2-acetolactate is common to the pathway to (R,R)-2,3-butanediol in the mixed-acid fermentation of pyruvate. The first step of the pathway, acetolactate synthase, requires thiamine pyrophosphate and two molecules of pyruvate, one of which is decarboxylated to the acetoacetate adduct, 2-(alpha-hydroxyethyl)-ThPP.',\n",
       "  None),\n",
       " ('GenProp0164',\n",
       "  'Leucine biosynthesis from pyruvate and acetyl-CoA',\n",
       "  'PATHWAY',\n",
       "  'The branched amino acid leucine is biosynthesized from the central intermediary metabolites pyruvate and acetyl-CoA via a seven step pathway. Four of these steps are shared with the valine and isoleucine pathways, although specialized versions of these enzymes may exist in certain species which have a narrowed substrate range or amino acid-specific regulatory mechanisms. The product of the third step in this pathway, 2-oxoisovalerate is a branchpoint metabolite required for the biosynthesis of Coenzyme A, branched-chain fatty acids and valine. Similarly, the product of the first step, (S)-2-acetolactate is common to the pathway to (R,R)-2,3-butanediol in the mixed-acid fermentation of pyruvate. The first step of the pathway, acetolactate synthase, requires thiamine pyrophosphate and two molecules of pyruvate, one of which is decarboxylated to the acetoacetate adduct, 2-(alpha-hydroxyethyl)-ThPP. The acetyl-CoA required for the fourth step, 2-isopropylmalate synthase, may also derive from this ThPP adduct by oxidation.',\n",
       "  None),\n",
       " ('GenProp0165',\n",
       "  'MinCDE system',\n",
       "  'SYSTEM',\n",
       "  'The MinCDE proteins are organized into extended membrane-associated coiled structures that wind around the cell between the two poles. These structures, and changes (pole-to-pole oscillations) in the distribution of the MinCDE proteins within them over time, are required for the proper placement of the cell division septum at midcell [1,2].',\n",
       "  None),\n",
       " ('GenProp0166',\n",
       "  'MreBCD system',\n",
       "  'SYSTEM',\n",
       "  'The bacterial MreBCD proteins are involved in rod shape determination and have a postulated role in the regulation of septum-specific peptidoglycan synthesis [1].Members of the actin-like MreB family of proteins localize as a helical filament in bacteria and are important for determining cylindrical cell shape. Recent results show that new cell wall biosynthesis occurs along a helical track dependent on one of these actin homologs, providing new insights into bacterial cell growth, division and shape [2].MreC and MreD are membrane-associated proteins.MreC is an essential gene in B. subtilis; cells lacking sufficient quantities of MreC undergo morphological changes, namely, swelling and twisting of the cells, which is followed by cell lysis [3].MreC and MreD may be involved in the regulation of cell-shape-related protease expression [4]. There is evidence of the existence of a separate rod-shape determining system in rod-shaped bacteria lacking the MreBCD system [5].',\n",
       "  None),\n",
       " ('GenProp0167',\n",
       "  'Cellular growth, organization and division',\n",
       "  'CATEGORY',\n",
       "  'Processes which determine, regulate or alter the physical size, shape or organization of the cell.',\n",
       "  None),\n",
       " ('GenProp0168',\n",
       "  'Glycogen system',\n",
       "  'SYSTEM',\n",
       "  'Glycogen and starch are related glucose storage polysaccharides. Alpha 1,4 glucan linkages predominate, with some branching using alpha 1,6 linkages. The relative level of branching determines whether the substance is called glycogen or starch. In bacteria and plants, the precursor is ADP-glucose. In animals, the precursor isUDP-glucose. Different forms of glycogen synthase are used for the two. This property includes the enzyme phosphorylase (glycogen phosphorylase or starch phosphorylase, depending on context) necessary to release glucose-1-phosphate from the polysaccharide.',\n",
       "  None),\n",
       " ('GenProp0170',\n",
       "  'Coenzyme PQQ biosynthesis',\n",
       "  'SYSTEM',\n",
       "  'Coenzyme PQQ (coenzyme pyrrolo-quinoline-quinone) is a cofactor of certain alcohol dehydrogenases and glucose dehydrogenases. PQQ biosynthesis is associated with an operon that includes PqqA, a small protein proposed to act as a molecular precursor. At least 6 genes are located within the PQQ-operon, all of which are required for PQQ biosynthesis, although in some cases the exact mechanism of the gene product is not known [1].',\n",
       "  None),\n",
       " ('GenProp0171',\n",
       "  'Coenzyme A biosynthesis from pantothenate',\n",
       "  'PATHWAY',\n",
       "  'This property represents a 5-step pathway for the biosynthesis of coenzyme A (CoA) from pantothenate. Most species produce panthothenate but some must import it. The first step described here is a sodium/pantothenate symporter. The first required step, catalyzed by pantothenate kinase, is carried out by at least two different known protein families. One is found in most bacteria, including E. coli (TIGR00554). Another is present in eukaryotes and in a few Gram-positive bacteria (e.g. Staphylococcus aureus, Bacillus anthracis) (TIGR00555). The enzyme, if present, has not been identified in a number of species. The second and third enzymes are usually found together in a multifunctional protein, phosphopantothenoylcysteine decarboxylase/phosphopantothenate--cysteine ligase, once thought to affect DNA metabolism as well. In Streptococcus and allies, and some other species, the enzymes are encoded in tandem genes rather than fused. CoaD and CoaE are, respectively, pantetheine-phosphate adenylyltransferase and dephospho-CoA kinase [1].',\n",
       "  None),\n",
       " ('GenProp0172',\n",
       "  'Potassium-transporting ATPase KdpFABC',\n",
       "  'SYSTEM',\n",
       "  'The Kdp complex is a high affinity P-type ATPase potassium ion transport system, found widely distributed among prokaryotes. Regulated by increases in osmolarity and/or reduction in external potassium, Kdp is the only K+ transport system whose expression is regulated by environmental conditions [1]. Kdp is composed of four membrane-bound subunits responsible for potassium ion binding and transport, ATP hydrolysis, and complex assembly and stability [2]. Not defined by this property are the sensor kinase and response regulator components, which once stimulated by specific environmental signals, induce expression of the adjacent Kdp gene cluster [1,2].',\n",
       "  None),\n",
       " ('GenProp0176',\n",
       "  'TRAP-T (tripartite ATP-independent periplasmic transporters) family transporters',\n",
       "  'SYSTEM',\n",
       "  'This property represents the presence of three-subunit transporters related to the DctPQM system for dicarboxylate transport in Rhodobacter capsulatus. These are secondary transporters, energized by ion gradients, rather than primary transporters powered by ATP directly [1]. The acronym TRAP, or TRAP-T, stands for Tripartite ATP-independent Periplasmic Transporters. The characteristic architecture includes an integral membrane protein with 12 predicted transmembrane (TM) regions (e.g.DctM) and another with 4 predicted TM regions (e.g. DctQ). These two are fused in some species [2]. Also present is an extracytoplasmic solute receptor (ESR). In addition to C4 dicarboxylates, substrates moved by related transporters appear to include mannitol, aspartate derivative ectoine, and other as yet uncharacterized solutes. Some bacteria, such as Bradyrhizobium japonicum, may contain over a dozen such transporters.',\n",
       "  None),\n",
       " ('GenProp0178',\n",
       "  'Glycine cleavage system',\n",
       "  'SYSTEM',\n",
       "  'The glycine cleavage system (GCS), also known as the glycine decarboxylase system (GDC) consists of a set of four enzymes, which are involved in a multi-step reaction to catalyse the cleavage of glycine molecules. The constituents enzymes are, P protein, H protein, T protein, and L protein (dihydrolipoamide dehydrogenase)[1].',\n",
       "  None),\n",
       " ('GenProp0179',\n",
       "  'Protein transport',\n",
       "  'CATEGORY',\n",
       "  'The translocation of proteins across biological membranes or other layers.',\n",
       "  None),\n",
       " ('GenProp0180',\n",
       "  'Small molecule transport',\n",
       "  'CATEGORY',\n",
       "  'The translocation of small molecules and ions across biological membranes or other layers.',\n",
       "  None),\n",
       " ('GenProp0182',\n",
       "  'Reduction of oxidized methionine',\n",
       "  'GUILD',\n",
       "  'Methionine, free or in proteins, is susceptible to oxidation damage. R- and S-conformation enantiomers of methionine sulfoxide can be repaired by different reductases. MsrB (TIGR00357) was previously designated PilB because of altered cytoadherence in a mutant. MsrA and MsrB are sometimes expressed as a fusion protein.',\n",
       "  None),\n",
       " ('GenProp0183',\n",
       "  'dTDP-L-rhamnose biosynthesis from dTDP-4-dehydro-L-rhamnose',\n",
       "  'METAPATH',\n",
       "  'L-rhamnose is a common component of the bacterial capsule and/or cell wall, and is important in bacterial survival during infection. dTDP-L-rhamnose is the precursor of L-rhamnose[1,2]. dTDP-L-rhamnose is synthesized from glucose-1-phosphate and deoxythymidine triphosphate (dTTP)\\xa0via\\xa0a four-step enzymatic pathway. The four enzymes are: glucose-1-phosphate thymidyltransferase (also known as dTDP- glucose pyrophosphorylase, gene name rlmA or rfbA); dTDP-D-glucose 4,6-dehydratase (gene name rmlB or rfbB); dTDP-4-dehydrorhamnose 3,5-epimerase (also known as dTDP-6-deoxy-D-xylo-4-hexulose 3,5-epimerase, gene name rmlC or rfbC); and dTDP-4-dehydrorhamnose reductase (also known as dTDP-6-deoxy-L-lyxo-4-hexulose reductase (gene name rmlD or rfbD)[1,2]. This property refers to the final step in this pathway, the conversion of dTDP-4-dehydro-L-rhamose to dTDP-L-rhamnose.',\n",
       "  None),\n",
       " ('GenProp0184',\n",
       "  'Cofactor biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The biosynthesis of molecules utilized by enzymes to aid in chemical transformations.',\n",
       "  None),\n",
       " ('GenProp0185',\n",
       "  'Nucleotide biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The biosynthesis of the components pyridine and pyrimidine of DNA and RNA.',\n",
       "  None),\n",
       " ('GenProp0186',\n",
       "  'Storage and structural polymer biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The biosynthesis of polymers not including standard protein and polynucleotide biosynthesis.',\n",
       "  None),\n",
       " ('GenProp0187',\n",
       "  \"Pyrimidine (uridine-5'-phosphate) de novo biosynthesis\",\n",
       "  'PATHWAY',\n",
       "  'De novo pyrimidine biosynthesis procedes from HCO3 (bicarbonate), an NH2 group from glutamine or free ammonia, and aspartate. Firstly, carbamoyl-phosphate synthase produces carbamoyl phosphate, which is then condensed with L-aspartate to produce a ring compound, N-carbomoyl-L-aspartate, which is converted in two more steps into orotate. The ribose moiety is then added from alpha-D-5-P-ribosylpyrophosphate (PRPP), which is also a precursor of purines, histidine, and tryptophan. The resulting orotidine monophosphate is decarboxylated to uridine monophosphate (UMP), a pyrimidine nucleotide from which other pyrimidines are made.The few species lacking pyrimidine de novo biosynthesis include some members of the Chlamydiae, Mycoplasmatales, Rickettsiales, and Spirochaetes.',\n",
       "  None),\n",
       " ('GenProp0188',\n",
       "  'GatABC aspartyl/glutamyl-tRNA(Asn/Gln) amidotransferase complex',\n",
       "  'PATHWAY',\n",
       "  'The GatABC heterotrimer acts as an alternative to direct aminoacylation of tRNA(Gln), tRNA(Asn), or both with its respective amino acid. Rather, the tRNA may be misacylated with Glu in the place of Gln, or Asp in the place of Asn. The phylogenetic ranges over which GatABC orthologs act on Glu-tRNA(Gln), Asp-tRNA(Asn), or both are not fully worked out. It is both in Chlamydia, the Thermus-Deinococcus lineage, and Acidithiobacillus ferrooxidans. In the Archaea, GatDE act on Glu-tRNA(Gln) while GatABC act on Asp-tRNA(Asn). In Lactobacillus and possibly all Firmicutes, GatABC acts on Glu-tRNA(Gln) only. Note that the amidotransferase reaction is a mechanism not only to charge a tRNA with Asn or Gln, but to complete the synthesis of that amino acid.',\n",
       "  None),\n",
       " ('GenProp0189',\n",
       "  'tRNA-Gln direct aminoacylation',\n",
       "  'PATHWAY',\n",
       "  'The tRNA for Gln is produced, in different species, in two different ways. In some bacteria (including E. coli) and in the eukaryotic cytosol, the tRNA for glutamine is charged with glutamine. In the majority of sequenced prokaryotes however, tRNA(gln) initially is mischarged with glutamate (Glu). Glu is subsequently amidated to Gln while on the tRNA(gln), as by GatABC in Bacillus subtilis, resulting in Gln biosynthesis from Glu as well as tRNA aminoacylation. This property describes the pathway of direct Gln--tRNA ligase activity, performed by a single protein, glutaminyl-tRNA synthetase.',\n",
       "  None),\n",
       " ('GenProp0190',\n",
       "  'Phosphate ABC transporter (pstSCAB-phoU)',\n",
       "  'SYSTEM',\n",
       "  'This property represents an ABC-type transporter for inorganic phosphate P(i), consisting of genes related to the pstSCAB genes in Escherichia coli. In E. coli,pstSCAB represents a high-affinity transporter and is induced by low external P(i) concentrations; a constitutive, low affinity transporter PitA, is also found[1]. Included in this property is PhoU, a regulatory protein of unknown mechanism, generally found adjacent to the pstSCAB genes.',\n",
       "  None),\n",
       " ('GenProp0191',\n",
       "  'Sulfate/thiosulfate ABC transporter',\n",
       "  'SYSTEM',\n",
       "  'This property describes an ATP-binding cassette (ABC) transporter complex for sulfate and some related anions. Gene symbols for these proteins reflect a role of sulfate import in cysteine biosynthesis. The complex consists of four proteins, an ATP-binding protein CysA, two homologous permeases CysT and CysW, and a binding protein. In the model bacterium Escherichia coli, two binding proteins designated Sbp and CysP are present, and appear to facilitate transport of both sulfate and thiosulfate ions [1]. Unlike the case for phosphate uptake, ABC transport appears to be the only mechanism for sulfate uptake. An older report finds the sulfate transporter in E. coli is also the sole transporter for selenate and selenite [2], and a review on the molybdate ABC transporter notes that, in absense of such a transporter, the sulfate transporter can take up molybdate [3].',\n",
       "  None),\n",
       " ('GenProp0192',\n",
       "  'Molybdate ABC transporter',\n",
       "  'SYSTEM',\n",
       "  'The high-affinity molybdate ABC transporter imports molybdate for the biosythesis of cofactors such as molybdopterin. Three structural components of this typical ABC transporter are the ATPase subunit, permease, and binding protein. These structural components appear in three gene clusters. A molybdate transport repressor is included as a non-required part of the system. It is noted that molybdate can also be transported by some sulfate/thiosulfate ABC transporters and non-specific anion transporters [1].',\n",
       "  None),\n",
       " ('GenProp0193',\n",
       "  'Lysine biosynthesis via alpha-aminoadipate (AAA pathway)',\n",
       "  'PATHWAY',\n",
       "  'Lysine biosynthesis in fungi has been characterized and begins with the condensation of 2-oxoglutarate and acetyl-CoA to homocitrate and continues through the distinctive intermediate, alpha-aminoadipate. This pathway is distinct in every respect from the diaminopimelate pathway commonly found in bacteria and animals.Recently, an alpha-aminoadipate pathway closely related to the fungal version has been characterized in Thermus thermophilus [1] and appears to be widely distributed among the archaea.',\n",
       "  None),\n",
       " ('GenProp0196',\n",
       "  'Sulfur metabolism',\n",
       "  'CATEGORY',\n",
       "  'Processes handling and interconverting sulfur-containing molecules. This does not include cysteine and methionine-containing proteins.',\n",
       "  None),\n",
       " ('GenProp0197',\n",
       "  'Protein biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The formation of proteins from amino acid building blocks.',\n",
       "  None),\n",
       " ('GenProp0198',\n",
       "  'RuvABC Holliday junction complex',\n",
       "  'SYSTEM',\n",
       "  'RuvA, RuvB and RuvC proteins together act on Holliday junctions for homologous recombination and DNA repair. RuvAB is the branch migration complex and RuvC the resolvase [1]. The system is strictly bacterial; proteins with little or no detectable sequence similarity process Holliday junctions in Archaea, eukaryotes, and some phage. A variety of Holliday junction-cleaving (Hjc) enzymes have been found in the Archaea [2]. Holliday junction resolvases, including RuvC, Hjc and Hje, CCE1 (yeast) and A22R (poxviruses) are reviewed by Sharples [3]. RuvC is replaced by RecU in firmicutes and mollicutes.',\n",
       "  None),\n",
       " ('GenProp0199',\n",
       "  'Lysine biosynthesis',\n",
       "  'METAPATH',\n",
       "  'The basic amino acid lysine may be synthesized by one of two types of pathways in prokaryotes, diaminopimelate (DAP) pathways and the alpha-aminoadipate (AAA) pathway. DAP pathways have several variants including the dehydrogenase (reductive) pathway which requires ammonia and NADPH, and the acetylated and succinylated pathways which do not. A non-reductive pathway which uses neither acetate or succinate protecting groups is found in plants but has not been observed in prokaryotes. The AAA pathway has a variant found in fungi and plants which has not been observed in prokaryotes.',\n",
       "  None),\n",
       " ('GenProp0201',\n",
       "  'SMC-ScpA-ScpB complex',\n",
       "  'SYSTEM',\n",
       "  'scpA and scpB (segregation and condensation) are proposed [1,2] to act with SMC (structural maintenance of chromosome) in chromosome organization. SMC is not universal, but is found in Eukaryotes, Archaea, and Bacteria. It is homodimeric in Archaea and Bacteria. A mutation study [2] has shown a role for scpA in DNA repair in addition to its role in chromosome condensation and segregation. Proteins that act with SMC may be termed kleisins. Different SMC/kleisin complexes are termed cohesins and condensins. ScpA and ScpB are clustered in most bacteria. SMC and scpA tend to be clustered in the archaea.',\n",
       "  None),\n",
       " ('GenProp0202',\n",
       "  'Nitrogen metabolism',\n",
       "  'CATEGORY',\n",
       "  'Processes handling and interconverting nitrogen-containing molecules.',\n",
       "  None),\n",
       " ('GenProp0207',\n",
       "  'Lipoprotein localization system lolABCDE',\n",
       "  'SYSTEM',\n",
       "  'Bacterial lipoproteins have a modified amino terminal Cys residue with a covalently attached lipid moiety that serves as a membrane anchor. The lol system is responsible for moving lipoproteins from the periplasmic side of the inner membrane to the periplasmic side of the outer membrane. LolD is homologous to the ATP-binding cassette (ABC) transport family [1,2]. LolC and LolE are homologous to each other (in species that have both rather than just one) and have several membrane-spanning segments, much like ABC transport permeases[3]. However, the LolCDE complex differs from other ABC transporters in that is does not move a substrate across a membrane. Instead, it detaches lipoproteins from the inner membrane. LolA and LolB act to chaperone lipoproteins, including LolB itself, to the outer membrane. This system is less broadly distributed than the lgt/lsp/lnt system for lipoprotein N-terminal modification.',\n",
       "  None),\n",
       " ('GenProp0209',\n",
       "  'Sec system preprotein translocase',\n",
       "  'SYSTEM',\n",
       "  'This property describes the bacterial preprotein translocase Sec[1]. The analogous process occurs in eukaryotes and archaea (little studied in the archaea), but the component proteins are different enough not to be described by the same set of hidden Markov models. SecB is a cytosolic export-specific chaperone in E. coli and some related species, but is not broadly distributed. SecA, the ATPase that powers the translocase, is a peripheral membrane protein. SecY, SecE, and SecG form the heterotrimeric translocase, although SecG is not essential at some temperatures. SecD, SecG, and YajC form an accessory complex that may prevent backsliding of proteins being translocated, and may not be essential proteins.',\n",
       "  None),\n",
       " ('GenProp0216',\n",
       "  'RecBCD pathway',\n",
       "  'SYSTEM',\n",
       "  'This property describes the presence of genes for the RecBCD protein complex, which participates in at least two different processes. As exodeoxyribonuclease V, it degrades linear double-stranded DNA, as might occur by cleavage of exogenous DNA by restriction systems. It also acts to repair double-stranded breaks in DNA by homologous recombination [1]; genera that lack this complex tend to show stronger conservation of gene order between species[2]. The RecBCD heterotrimer interacts with instances of the DNA sequence GCTGGTGG in E. coli, called Chi, which are recombinational hotspots. RecBCD promotes loading the strand exchange protein RecA onto chi-containing DNA[3]. Double-stranded breaks are processed by AddAB rather than by RecBCD in the Firmicutes and in alpha-Proteobacteria.',\n",
       "  None),\n",
       " ('GenProp0217',\n",
       "  '5-aminolevulinate biosynthesis (glutamate pathway)',\n",
       "  'PATHWAY',\n",
       "  'This pathway for the biosynthesis of 5-aminolevulinate, a porphyrin precursor, is known as the glutamate or C5 pathway and is found in most bacteria. A distinctive feature of this pathway is the utilization of an amino acid-charged tRNA as an enzymatic substrate.An alternative pathway from succinyl-CoA and glycine utilizing 5-aminolevulinate synthase is found in the alphaproteobacteria and eukaryotic mitochondria[1,2,3].',\n",
       "  None),\n",
       " ('GenProp0223',\n",
       "  '5-aminolevulinate biosynthesis from succinyl-CoA and glycine',\n",
       "  'PATHWAY',\n",
       "  'This property is an alternative to the biosynthesis of 5-aminolevulinate through the glutamate pathway. 5-aminolevulinate is used in the biosynthesis of uroporphyrinogen III and farther downstream, in the biosynthesis of heme[1]. This enzyme appears restricted to the alpha Proteobacteria and mitochondrial derivatives.',\n",
       "  None),\n",
       " ('GenProp0227', 'Excision repair', 'CATEGORY', None, None),\n",
       " ('GenProp0228', 'Direct repair', 'CATEGORY', None, None),\n",
       " ('GenProp0231',\n",
       "  '4-hydroxyphenylacetate degradation',\n",
       "  'PATHWAY',\n",
       "  '4-hydroxyphenylacetic acid, a degradation product of tyrosine via one of several routes, of phenylalanine via phenylacetate, and of environmental DDT is metabolized in this pathway to the common metabolites succinate and pyruvate via homoprotocatechuate[1]. 4-hydroxyphenylacetic acid may also be found in the environment and transported into the cell.The genes encoding the components of this pathway are often organized into apparent operons which include regulatory elements as well as a variety of transport systems.A number of genomes show partial evidence of the pathway based on homologs of the well-characterized enzymes from other species.Often, promising candidates for the missing steps are found in the vicinity. These are suggestive of enzymes which have been adapted from ancestral sequences having a different function.Of course, the possibility exists that it is the entire pathway that has been modified in these species and it is other structurally related compounds that are being metabolized instead of 4-hydroxyphenylacetic acid.',\n",
       "  None),\n",
       " ('GenProp0238',\n",
       "  '2-aminoethylphosphonate catabolism to acetaldehyde',\n",
       "  'PATHWAY',\n",
       "  'The enzyme 2-aminoethylphosphonate--pyruvate transaminase (AEPT) interconverts 2-aminoethylphosphonate plus pyruvate with 2-phosphonoacetaldehyde plus alanine. The enzyme phosphonoacetaldehyde hydrolase, a HAD superfamily hydrolase, breaks the C-P bond, yielding acetaldehyde plus inorganic phosphate[1]. In species that appear to encode a phosphonate ABC transporter, this property is likely to be present only when the large phosphonate operon proposed to encode a multisubunit C-P lyase, as found in Escherichia coli, is not found.',\n",
       "  None),\n",
       " ('GenProp0239', 'Propionyl-CoA catabolism', 'CATEGORY', None, None),\n",
       " ('GenProp0242',\n",
       "  'Phosphorus metabolism',\n",
       "  'CATEGORY',\n",
       "  'The transport and transformations of biomolecules containing the phosphorus atom.',\n",
       "  None),\n",
       " ('GenProp0243',\n",
       "  'ATP synthases',\n",
       "  'CATEGORY',\n",
       "  'Systems which generate ATP from ADP and inorganic phosphate by utilizing membrane potential or chemical gradients.',\n",
       "  None),\n",
       " ('GenProp0244',\n",
       "  'Chaperone system: DnaK-DnaJ-GrpE',\n",
       "  'SYSTEM',\n",
       "  'DnaK, DnaJ, and GrpE form a chaperone system. DnaK has ATPase activity. DnaJ and GrpE are co-chaperones for DnaK, that is, stimulate this activity[1]. This system is also called the hsp70 heat-shock system. Excluded from this system are the heat shock cognate (Hsc) homologs, which are associated specifically with iron-sulfur cluster biosynthesis (see GenProp0138, the iron-sulfur cluster assembly iscSUA-hscBA-fdx system). The heat shock repressor HrcA is found in about half of bacterial genomes with the DnaK-DnaJ-GrpE heat shock system, usually encoded upstream in a heat shock operon; it is included here as an optional component.',\n",
       "  None),\n",
       " ('GenProp0245',\n",
       "  'Chaperone system: GroEL/GroES',\n",
       "  'SYSTEM',\n",
       "  'GroEL and GroES comprise the bacterial group I chaperonin, a toroidal complex involved in folding or refolding proteins in the cytosol[1]. Archaea and Eukaryotes have instead (with a few rare cases of overlap) a group II chaperonin complex, usually called the thermosome in the Archaea and CCT in the Eukaryota. At least one member of the archaeal genus Methanosarcina, has both[2].',\n",
       "  None),\n",
       " ('GenProp0246',\n",
       "  'Chaperone system: thermosome/prefoldin',\n",
       "  'SYSTEM',\n",
       "  'The archaeal thermosome system is a group II chaperonin, closely related to the CCT complex of eukaryotes. Thermosome subunits show clear remote sequence similarity to GroEL of the bacterial group I chaperonin system[1,2]. Prefoldin, with alpha and beta subunits, works with the chaperonin. The number of distinct but homologous genes for subunits of the thermosome ranges from one to five in the archaea.',\n",
       "  None),\n",
       " ('GenProp0247',\n",
       "  'Protein metabolism',\n",
       "  'CATEGORY',\n",
       "  'The chemical reactions and physical changes involving the synthesis, modification and degradation of proteins.',\n",
       "  None),\n",
       " ('GenProp0248',\n",
       "  'Protein folding',\n",
       "  'CATEGORY',\n",
       "  'The transformation of polypeptide chains between the unfolded, misfolded and folded (active) states.',\n",
       "  None),\n",
       " ('GenProp0250',\n",
       "  'Hydroxyethylthiazole (HET) biosynthesis from 1-deoxy-D-xylulose-5-phosphate, Cysteine, and either Tyr or Gly',\n",
       "  'PATHWAY',\n",
       "  'This property describes de novo synthesis of the thiazole moiety of thiamine, hydroxyethylthiazole (HET). This requires more distinct proteins than that of the pyrimidine moeity, hydromethylpyrimidine (HMP). Phosphorylated forms of HET and HMP are connected by ThiE as thiamine phosphate and converted eventually to the active form thiamine pyrophosphate, TPP. HET biosynthesis requires the small ThiS protein, the C terminus of which becomes modified transiently by a sulfur atom from cysteine [1]. ThiG is required, along with either ThiH in species where tyrosine is a precursor or ThiO in species where glycine is a precursor. ThiF, although required biosynthetically, is not readily separated from MoeB, a molybdopterin biosynthesis protein, and so is not considered a required component. The ThiI gene is required, but also serves in formation of a tRNA modified site in species that do not make HET [2]. This property represents and alternative to HET uptake as a means to obtain the thiazole component for thiamine biosynthesis.',\n",
       "  None),\n",
       " ('GenProp0252',\n",
       "  'Thiamine/TPP ABC transporter ThiBPQ',\n",
       "  'SYSTEM',\n",
       "  'This property describes a three-component ABC transporter, ThiBPQ, for thiamine and/or thiamine derivatives. It is demonstrated experimentally in Salmonella typhimurium and E. coli[1], and is inferred from gene location next to genes for thiamine salvage enzymes in Brucella suis and Sinorhizobium meliloti. All known examples are in Proteobacteria.',\n",
       "  None),\n",
       " ('GenProp0253',\n",
       "  'Hydroxymethylpyrimidine pyrophosphate (HMP-P) biosynthesis from aminoimidazole ribotide',\n",
       "  'PATHWAY',\n",
       "  'Hydroxymethylpyrimidine pyrophosphate (HMP-PP) and hydroxyethylthiazole phosphate (HET-P) are the two components attached by ThiE to yield thiamine phosphate, which is subsequently converted to the active form thiamine pyrophosphate (TPP). HMP and HET may be imported and converted to HMP-PP and HET-P, respectively, as an alternative to de novo biosynthesis. This property reflects known steps of HMP-P de novo biosynthesis from the purine biosynthesis metabolite aminoimidazole ribotide (AIR), starting with the ThiC enzyme [1].',\n",
       "  None),\n",
       " ('GenProp0254',\n",
       "  'Thiamine pyrophosphate (TPP) biosynthesis from HMP-PP and HET-P',\n",
       "  'PATHWAY',\n",
       "  'Thiamine pyrophosphate (TPP) is an essential cofactor for several carbohydrate metabolism enzymes. Hydroxymethylpyrimidine pyrophosphate (HMP-PP) may be obtained by de novo synthesis, or by import or salvage of hydroxymethylpyrimidine followed by phosphorylation. Likewise, hydroxyethylthiazole phosphate (HET-P) may come from synthesis, import, or salvage. The two heterocylic compounds are ligated by ThiE to thiamine phosphate. Thiamine phosphate then undegoes either dephosphorylation followed by pyrophosphorylation or direct phosphorylation [1].',\n",
       "  None),\n",
       " ('GenProp0255',\n",
       "  '5-aminolevulinate biosynthesis',\n",
       "  'METAPATH',\n",
       "  '5-aminolevulinate, a heme precursor, may be biosynthesized by one of two pathways. The C5 (or Glutamate) pathway is more common [GenProp0217], while alphaproteobacteria utilize the enzyme 5-aminolevulinate synthase to make the molecule from succinyl-CoA and glycine [GenProp0223].',\n",
       "  None),\n",
       " ('GenProp0261',\n",
       "  'Thiamine pyrophosphate (TPP) de novo biosynthesis',\n",
       "  'METAPATH',\n",
       "  'This property describes de novo synthesis of teh essential cofactor thiamine pyrophosphate (TPP) [1]. The process consists of the synthesis of hydroxyethylthiazole phosphate (HET-P), the synthesis of hydroxymethylpyrimidine pyrophosphate (HMP-PP), and their ligation and conversion to TPP.',\n",
       "  None),\n",
       " ('GenProp0262',\n",
       "  'RNA polymerase, bacterial',\n",
       "  'SYSTEM',\n",
       "  \"This property describes the bacterial multisubunit DNA-directed RNA polymerase (EC 2.7.7.6). This property does not include sigma factors which are regulatory and varied, and so by definition this property describes the core enzyme rather than the holoenzyme. Subunit stochiometry is two alpha subunits to one each of beta and beta'. In Cyanobacteria and plastids, beta' is replaced by two proteins, one like the N-term of beta' (called gamma in Cyanobacteria and, confusingly, beta' in plastids) and one like the C-term of beta' (called beta' in Cyanobacteria and beta'' in plastids)[1].\",\n",
       "  None),\n",
       " ('GenProp0267',\n",
       "  'Osmolyte biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'Cells can resist dessication, high salt, and other osmotic stresses by modulating levels of osmolytes, compatible solutes that can exists in high concentrations without damaging cells. Some osmolytes may have secondary functions for carbon and energy storage molecules or for further biosynthetic reactions. Widely distributed compatible solutes induced by osmotic stress include trehalose, glucosylglycerol, sucrose, mannitol, glycine betaine and other betaines, ectoine and hydroxyectoine, glutamate, N-acetylglutaminylglutamine amide, etc.',\n",
       "  None),\n",
       " ('GenProp0297', 'Lipopolysaccharide biosynthesis', 'CATEGORY', None, None),\n",
       " ('GenProp0298', 'Fermentation', 'CATEGORY', None, None),\n",
       " ('GenProp0299', 'Fatty acid biosynthesis', 'CATEGORY', None, None),\n",
       " ('GenProp0300',\n",
       "  'Photosynthetic cofactor biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The biosynthesis of compounds which assist in the harvesting of light energy, such as chlorophyll and bacteriochlorophyll.',\n",
       "  None),\n",
       " ('GenProp0301',\n",
       "  'Protein degradation',\n",
       "  'CATEGORY',\n",
       "  'The breakdown of proteins and polypeptides into smaller pieces.',\n",
       "  None),\n",
       " ('GenProp0302',\n",
       "  'Protein repair',\n",
       "  'CATEGORY',\n",
       "  'The process of restoring a protein to its original state after damage by such things as oxidation or spontaneous decomposition of residues.',\n",
       "  None),\n",
       " ('GenProp0303',\n",
       "  'Detoxification',\n",
       "  'CATEGORY',\n",
       "  'The process of making deleterious compounds, whether exogenous or generated within the cell, less dangerous via the mechanisms of chemical conversion, transport or sequestration.',\n",
       "  None),\n",
       " ('GenProp0462', 'Response to stress', 'CATEGORY', None, None),\n",
       " ('GenProp0475', 'Unknown biological process', 'CATEGORY', None, None),\n",
       " ('GenProp0501', 'Metrics of biological environment', 'CATEGORY', None, None),\n",
       " ('GenProp0551', 'Regulatory systems', 'CATEGORY', None, None),\n",
       " ('GenProp0603',\n",
       "  'Lifestyle',\n",
       "  'CATEGORY',\n",
       "  'Genome Properties in this category describe the different modes, styles and strategies by which organisms live. Lifestyles describe distinguishing characteristics by which these organisms may be grouped, and generally will have an underlying metabolic basis.',\n",
       "  None),\n",
       " ('GenProp0630',\n",
       "  '2-oxoglutarate dehydrogenase system',\n",
       "  'SYSTEM',\n",
       "  'This enzyme system utilizes three components E1 (thiamine pyrophosphate-binding subunit, SucA), E2 (dihydrolipoamide succinyltransferase, SucB) and E3 (dihydrolipoamide dehydrogenase, LpdA) for the interconversion of 2-oxoglutarate (alpha-ketoglutarate) and succinyl-CoA. In addition to lipoate and thiamine pyrophosphate, FAD is also involved. In the course of the reaction CO2 is released and one molecule of NAD+ is converted to NADH. This reaction is an essential component of the TCA (Krebs) cycle [1].',\n",
       "  None),\n",
       " ('GenProp0631',\n",
       "  'Nitrogenase, Fe-only',\n",
       "  'SYSTEM',\n",
       "  'There are three homologous types of nitrogenase; the common molybdenum-containing, the alternative vanadium (V)-containing, and the alterantive iron (Fe)-only. This Genome Property represents the Fe-only nitrogenase system (which acts in nitrogen fixation) and its accessory proteins. Fe-only nitrogenase is composed of 2 copies each of an alpha, beta and delta subunit, forming a hexameric structure. The exact function of the accessory protein included in this property is not yet known [1].',\n",
       "  None),\n",
       " ('GenProp0632',\n",
       "  'Nitrogenase, V-containing',\n",
       "  'SYSTEM',\n",
       "  'There are three homologous types of nitrogenase; the common molybdenum-containing, the alternative vanadium (V)-containing, and the alterantive iron (Fe)-only. This Genome Property represents the V-containing nitrogenase, which acts in nitrogen fixation. V-containing nitrogenase is composed of three subunits encoded by three genes within an operon: subunit alpha (vnfD), subunit beta (vnfK) and subunit delta (vnfG) [1].',\n",
       "  None),\n",
       " ('GenProp0633',\n",
       "  'Nitrogenase, Mo-containing',\n",
       "  'SYSTEM',\n",
       "  'There are three distict types of nitrogenase system that exist in bacteria and archea, acting in nitrogen fixation. The most common type of nitrogenase contains molybdenum as well as iron (MoFe nitrogenase). Alternative nitrogenases with homologous alpha and beta subunits, but somewhat different sets of accessory proteins, use vanadium and iron, or just iron. All species studied so far with one or more alternative nitrogenases, also contain a MoFe nitrogenase. This genome property consists of the structural subunits of the MoFe nitrogenase [1].',\n",
       "  None),\n",
       " ('GenProp0723',\n",
       "  'Hydrogenase, nickel and F420-dependent',\n",
       "  'SYSTEM',\n",
       "  'This nickel hydrogenase reduces coenzyme F420 (8-hydroxy-5-deazaflavin) and is found exclusively, but not universally, in methanogenic archaea. It is lacking in at least four Methanosarcinales. F420 reducing hydrogenase (FRH) is composed of three subuints, aplpha, beta and gamma, which are encoded by the genes frhA, frhB, and frhG respectively. A forth gene in the same cluster, frhD, encodes a delta subunit, which is not part of the active FRH heterotrimer, but may be required for maturation [1].',\n",
       "  None),\n",
       " ('GenProp0731',\n",
       "  'Amino acid metabolism',\n",
       "  'CATEGORY',\n",
       "  'The chemical reactions and pathways involving amino acids (organic acids containing one or more amino substituents).',\n",
       "  None),\n",
       " ('GenProp0733',\n",
       "  'Amino acid salvage',\n",
       "  'CATEGORY',\n",
       "  'Any process which produces an amino acid from derivatives of it, without de novo synthesis.',\n",
       "  None),\n",
       " ('GenProp0734',\n",
       "  'Amino acid catabolism',\n",
       "  'CATEGORY',\n",
       "  'The chemical reactions and pathways resulting in the breakdown of amino acids (organic acids containing one or more amino substituents).',\n",
       "  None),\n",
       " ('GenProp0770', 'Siderophore biosynthesis', 'CATEGORY', None, None),\n",
       " ('GenProp0786',\n",
       "  'Lysine biosynthesis via diaminopimelate (DAP), succinylated branch',\n",
       "  'PATHWAY',\n",
       "  'There are four known variations (branches) of the lysine biosynthesis pathway via diaminopimelate (DAP). The variations occur in the steps converting L-2,3,4,5-tetrahydrodipicolinate (THDPA) to LL-2,6-diaminopimelate. This property refers to the succinylated branch of this pathway. This branch acylates THDPA with succinyl-CoA to generate N-succinyl-ll-2-amino-6-ketopimelate and forms meso-DAP by subsequent transamination, desuccinylation, and epimerization. This branch is utilized by proteobacteria as well as many firmicutes and actinobacteria [1].',\n",
       "  None),\n",
       " ('GenProp0787',\n",
       "  'Lysine biosynthesis via diaminopimelate (DAP), acetylated branch',\n",
       "  'PATHWAY',\n",
       "  'There are four known variations (branches) of the lysine biosynthesis pathway via diaminopimelate (DAP). The variations occur in the steps converting L-2,3,4,5-tetrahydrodipicolinate (THDPA) to LL-2,6-diaminopimelate. This property refers to the acetylated branch of this pathway which generates N-acetyl intermediates. This branch is limited to certain Bacillus species [1].',\n",
       "  None),\n",
       " ('GenProp0788',\n",
       "  'Lysine biosynthesis via diaminopimelate (DAP) utilizing ammonia and NADPH',\n",
       "  'METAPATH',\n",
       "  'This pathway for the biosynthesis of L-lysine differs from the more common diaminopimelate pathway by the action of diaminopimelate dehydrogenase (ddh). This enzyme requires both ammonia as a nitrogen source (instead of an amino acid via a transaminase) and reductive power in the form of NADPH. Additionally, this pathway does not require the acetyl/succinyl protecting group transformations found in the standard pathway, nor does it require the action of diaminopimelate epimerase. This enzyme has been best characterized in Corynebacterium glutamicum and Bacillus sphaericus. One should note that the ddh enzyme is shown acting on the compound L-2-amino-6-oxoheptanedioate in EC and KEGG representations. This is merely the ring-opened form of 2,3,4,5-tetrahydropicolinate with which it is in equilibrium, and upon which the acyltransferases of the common pathway act.',\n",
       "  None),\n",
       " ('GenProp0791',\n",
       "  'Coenzyme F420 biosynthesis',\n",
       "  'METAPATH',\n",
       "  'Coenzyme F420 is a hydride carrier cofactor functioning in methanogenesis [1]. The chromophore of coenzyme F420 is 7,8-didemethyl-8-hydroxy-5-deazaflavin (otherwise known as FO), which is also used in a cyanobacterial DNA photoreactivating enzyme. Modification of FO to create coenzyme F420 includes the GTP-dependent addition of two glutamate residues in Methanococcus jannaschii [2] and five or six in Mycobacterium bovis [3]. CofF, which caps the gamma-glutamyl tail after two glutamates, occurs only the archaea.',\n",
       "  None),\n",
       " ('GenProp0792',\n",
       "  '7,8-didemethyl-8-hydroxy-5-deazariboflavin (FO) biosynthesis',\n",
       "  'PATHWAY',\n",
       "  '7,8-didemethyl-8-hydroxy-5-deazariboflavin, or FO, is the chromophore of coenzyme F420. Several enzymatic steps convert FO to F420 (see GenProp0791). FO is also the chromophore of a cyanobacterial DNA photoreactivating enzyme [1]. The two key enzymes of this pathway, encoded by CofG and CofH in Methanocaldococcus jannaschii, are encoded by fusion proteins in certain lineages. The fusion protein in Mycobacterium smegmatis is designated FbiC (F420 BIosynthesis C). The precursors are a pyrimidine intermediate from riboflavin biosynthesis and the 4-hydroxyphenylpyruvate precursor of tyrosine, so expressing FbiC in Escherichia coli is sufficient for FO biosynthesis [2].',\n",
       "  None),\n",
       " ('GenProp0795',\n",
       "  'Carbohydrate biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'Processes related to the biosynthesis and modification of sugars (saccharides). These may then be utilized in central metabolism, the modification of cellular stuctures and/or incorporated into complex polymers.',\n",
       "  None),\n",
       " ('GenProp0816',\n",
       "  'Nucleoside catabolism/utilization pathways',\n",
       "  'CATEGORY',\n",
       "  None,\n",
       "  None),\n",
       " ('GenProp0817',\n",
       "  'Nucleotide catabolism/utilization pathways',\n",
       "  'CATEGORY',\n",
       "  None,\n",
       "  None),\n",
       " ('GenProp0819',\n",
       "  'RNA metabolism',\n",
       "  'CATEGORY',\n",
       "  'The chemical reactions and physical changes involving RNA (ribonucleaic acid), one of the two main types of nucleic acid. RNA is a long unbranched macromolecule, formed from one strand of linked ribonucleotides.',\n",
       "  None),\n",
       " ('GenProp0820',\n",
       "  'Iron metabolism',\n",
       "  'CATEGORY',\n",
       "  'Properties in this catergory cover processes that import, export and transform the chemical state of the iron atom.',\n",
       "  None),\n",
       " ('GenProp0829',\n",
       "  'Menaquinone biosynthesis via futalosine',\n",
       "  'PATHWAY',\n",
       "  'This pathway for the biosynthesis of menaquinone begins with chorismate, and progresses via the intermediate futalosine. This is completely separate to the well-known E. coli menaquinone synthesis pathway (GenProp0058), which also starts with chorismate [1].',\n",
       "  None),\n",
       " ('GenProp0836',\n",
       "  'Menaquinone biosynthesis',\n",
       "  'METAPATH',\n",
       "  'Menaquinone may be synthesized by one of two independent pathways. The better known pathway, which includes MenA, MenB, etc., as found in Escherichia coli, passes through the intermediate SEPHCHC (see GenProp0058). An alternate pathway, found in both Helicobacter pylori and Streptomyces coelicolor, involves the intermediate futalosine (see GenProp0829).',\n",
       "  None),\n",
       " ('GenProp0838',\n",
       "  'Lipid biosynthesis',\n",
       "  'CATEGORY',\n",
       "  'The chemical reactions and pathways resulting in the formation of lipids (compounds soluble in an organic solvent but not, or sparingly, in an aqueous solvent). These include but are not limited to: isoprenoids, steroids, fatty acids, membrane lipids and lipopolysaccharides.',\n",
       "  None),\n",
       " ('GenProp0864',\n",
       "  'Response to external environment',\n",
       "  'CATEGORY',\n",
       "  'Mechanisms by which cells sense the external environment and alter their motion, behavior, shape, metabolism or other properties in response to those conditions.',\n",
       "  None),\n",
       " ('GenProp0865', 'Carbon fixation', 'CATEGORY', None, None),\n",
       " ('GenProp0866',\n",
       "  'Carbon metabolism',\n",
       "  'CATEGORY',\n",
       "  'Processes involved in the handling and processing of small, primarily 1-carbon compounds. This category includes carbon fixation, C1-compound interconversion and excretion processes.',\n",
       "  None),\n",
       " ('GenProp0873', 'Ion-motive antiporter cycles', 'CATEGORY', None, None),\n",
       " ('GenProp0874',\n",
       "  'Response to acidic (low pH) environment',\n",
       "  'CATEGORY',\n",
       "  None,\n",
       "  None),\n",
       " ('GenProp0950',\n",
       "  'dTDP-4-dehydro-6-deoxy-alpha-D-glucose biosynthesis from glucose-1-phosphate',\n",
       "  'PATHWAY',\n",
       "  'dTDP-4-dehydro-6-deoxy-alpha-D-glucose is an intermediate in the biosynthesis of L-rhamnose and D-fucosamine derivatives. L-rhamnose is a common component of the bacterial capsule and/or cell wall, and is important in bacterial survival during infection. dTDP-L-rhamnose is the precursor of L-rhamnose[1,2]. dTDP-L-rhamnose is synthesized from glucose-1-phosphate and deoxythymidine triphosphate (dTTP) via a four-step enzymatic pathway. The four enzymes are: glucose-1-phosphate thymidyltransferase (also known as dTDP-glucose pyrophosphorylase, gene name rlmA or rfbA); dTDP-D-glucose 4,6-dehydratase (gene name rmlB or rfbB); dTDP-4-dehydrorhamnose 3,5-epimerase (also known as dTDP-6-deoxy-D-xylo-4-hexulose 3,5-epimerase, gene name rmlC or rfbC); and dTDP-4-dehydrorhamnose reductase (also known as dTDP-6-deoxy-L-lyxo-4-hexulose reductase (gene name rmlD or rfbD)[1,2]. This property refers to the first two steps of this pathway.',\n",
       "  None),\n",
       " ('GenProp1004',\n",
       "  'Light-independent protochlorophyllide reductase',\n",
       "  'SYSTEM',\n",
       "  'Protochlorophyllide reductase is the final step in the biosynthesis of chlorophyllide a from protoporphyrin IX. This genome property represents the three-component, light-independent protochlorophyllide reductase. There is also a separate light-dependent protochlorophyllide reductase which is not covered by this property.',\n",
       "  None),\n",
       " ('GenProp1032',\n",
       "  'dTDP-4-dehydro-L-rhamnose biosynthesis from dTDP-4-dehydro-6-deoxy-alpha-D-glucose',\n",
       "  'METAPATH',\n",
       "  'L-rhamnose is a common component of the bacterial capsule and/or cell wall, and is important in bacterial survival during infection. dTDP-L-rhamnose is the precursor of L-rhamnose[1,2]. dTDP-L-rhamnose is synthesized from glucose-1-phosphate and deoxythymidine triphosphate (dTTP) via a four-step enzymatic pathway. The four enzymes are: glucose-1-phosphate thymidyltransferase (also known as dTDP-glucose pyrophosphorylase, gene name rlmA or rfbA); dTDP-D-glucose 4,6-dehydratase (gene name rmlB or rfbB); dTDP-4-dehydrorhamnose 3,5-epimerase (also known as dTDP-6-deoxy-D-xylo-4-hexulose 3,5-epimerase, gene name rmlC or rfbC); and dTDP-4-dehydrorhamnose reductase (also known as dTDP-6-deoxy-L-lyxo-4-hexulose reductase (gene name rmlD or rfbD)[1,2]. This property refers to the third step in this pathway, the conversion of dTDP-4-dehydro-6-deoxy-alpha-D-glucose to dTDP-4-dehydro-L-rhamnose. dTDP-4-dehydro-L-rhamnose is a precursor of both dTDP-L-rhamnose and dTDP-6-deoxy-L-talose.',\n",
       "  None),\n",
       " ('GenProp1048',\n",
       "  'Quorum-sensing, biofilm formation, and developmental systems',\n",
       "  'CATEGORY',\n",
       "  None,\n",
       "  None),\n",
       " ('GenProp1067',\n",
       "  'Defense systems',\n",
       "  'CATEGORY',\n",
       "  'This category includes restriction systems, CRISPR, abortive infection systems, and other defenses against invasive nucleic acids.',\n",
       "  None),\n",
       " ('GenProp2012',\n",
       "  '19S proteasome cap',\n",
       "  'SYSTEM',\n",
       "  'The 20S proteasome is converted to the 26S proteasome by addition of the 19S cap.This is a complex of eighteen proteins, six AAA-ATPases and fourteen that are not ATPases.Non-ATPase regulatory subunit 14 is a metallo-dependent deubiquitinating enzyme from peptidase family M67.Non-ATPase regulatory subunit 4 binds ubiquitin. Ubiquitin receptor ADRM1 also binds ubiquitin and activates non-ATPase regulatory subunit 14, but is absent from the solved tertiary structures [1,2]. Non-ATPase regulatory subunits 5, 9, 10 and 11 are required for proteasome assembly. The 19S complex consists of a base, containing the six AAA-ATPases and three non-ATPase regulatory subunits, and a lid containing all the remaining non-ATPase regulatory subunits [4,5]. The functions of the 19S cap are to remove ubiquitin from targeted proteins, and to denature proteins prior to entry into the central cavity of the proteasome [3].',\n",
       "  None),\n",
       " ('GenProp2017',\n",
       "  'Gamma secretase',\n",
       "  'SYSTEM',\n",
       "  \"Gamma secretase is a complex found in cell membranes and initially identified as the peptidase responsible for the gamma, varepsilon and zeta cleavages of the transmembrane domains of Type I membrane proteins including amyloid precursor protein (APP) and Notch.Release of the amyloid beta (A-beta) peptide from APP is an important step in the development of Alzheimer's disease and forms the plaques that are characteristic of the pathology.Release of the A-beta peptide requires two proteolytic cleavages, cleavage by beta-secretase in the extracellular domain, and then cleavage at the gamma and epsilon sites by gamma secretase.The proteolytic component of gamma secretase is presenilin which is an aspartic endopeptidase [1].\",\n",
       "  None),\n",
       " ('GenProp2021',\n",
       "  'Apoptosome',\n",
       "  'SYSTEM',\n",
       "  'Apoptosis or programmed cell death is the mechanism by which cells in eukaryotes are killed. Cysteine endopeptidases known as caspases are activated and these deactivate cellular proteins, often by cleaving within interdomain regions, thereby separating binding domains of signal reception and transduction domains [1]. Caspases have a specificity for aspartyl bonds, which occur frequently in surface loops and interdomain regions of proteins [2]. Caspases involved in apoptosis are of two kinds: initiators and executors. Initiator caspases activate the executor caspases which cleave other proteins in the cell. There are at least four routes by which the initiator caspase precursors are activated. This Genome Property includes the apoptosome, which includes the precursor of initiator caspase 9 from the intrinsic (or mitochondrial) pathway [3]. When it binds cytochrome c and dATP, Apaf1 oligomerizes into a heptameric ring.',\n",
       "  None)]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gen_prop_data_rows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "with open('./gen_prop_data.tsv', 'w') as tsv_file:\n",
    "    csv_writer = csv.writer(tsv_file, delimiter='\\t')\n",
    "    for row in gen_prop_data_rows:\n",
    "        csv_writer.writerow(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
